{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b4254c18",
   "metadata": {
    "papermill": {
     "duration": 0.002191,
     "end_time": "2024-09-17T20:19:46.951760",
     "exception": false,
     "start_time": "2024-09-17T20:19:46.949569",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# Prepare data for Pipeline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "969bad8d",
   "metadata": {
    "papermill": {
     "duration": 0.001534,
     "end_time": "2024-09-17T20:19:46.956487",
     "exception": false,
     "start_time": "2024-09-17T20:19:46.954953",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "86cced95",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-10-14T14:33:37.863523Z",
     "start_time": "2022-10-14T14:33:14.145712Z"
    },
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:46.963166Z",
     "iopub.status.busy": "2024-09-17T20:19:46.962943Z",
     "iopub.status.idle": "2024-09-17T20:19:51.581987Z",
     "shell.execute_reply": "2024-09-17T20:19:51.581429Z"
    },
    "papermill": {
     "duration": 4.625148,
     "end_time": "2024-09-17T20:19:51.583257",
     "exception": false,
     "start_time": "2024-09-17T20:19:46.958109",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import yaml\n",
    "import logging\n",
    "import pandas as pd\n",
    "from Bio.Seq import Seq\n",
    "from qiime2 import Artifact, Visualization, Metadata\n",
    "from qiime2.plugins import demux\n",
    "from qiime2.plugins.cutadapt.methods import trim_paired, trim_single\n",
    "from qiime2.plugins.demux.methods import filter_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b5b63c7",
   "metadata": {
    "papermill": {
     "duration": 0.001622,
     "end_time": "2024-09-17T20:19:51.586739",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.585117",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "53befc19",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.594341Z",
     "start_time": "2022-06-28T18:50:41.591544Z"
    },
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.608653Z",
     "iopub.status.busy": "2024-09-17T20:19:51.608306Z",
     "iopub.status.idle": "2024-09-17T20:19:51.611871Z",
     "shell.execute_reply": "2024-09-17T20:19:51.611440Z"
    },
    "papermill": {
     "duration": 0.02444,
     "end_time": "2024-09-17T20:19:51.612800",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.588360",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "params_path = os.path.join('..', 'params', 'parameter-set-01.yaml')\n",
    "experiment_name = 'exp-01'\n",
    "base_dir = os.path.join('/', 'home', 'username', 'pipeline-dir')\n",
    "manifest_file = os.path.join(base_dir, 'data', 'raw', 'manifest', 'manifest.csv')\n",
    "replace_files = False\n",
    "trim = None\n",
    "metadata_file = None\n",
    "threads = 1\n",
    "demux_file = None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e3b6e058",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.616767Z",
     "iopub.status.busy": "2024-09-17T20:19:51.616536Z",
     "iopub.status.idle": "2024-09-17T20:19:51.619841Z",
     "shell.execute_reply": "2024-09-17T20:19:51.619419Z"
    },
    "papermill": {
     "duration": 0.006273,
     "end_time": "2024-09-17T20:19:51.620724",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.614451",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "base_dir = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/mnt/nupeb/rede-micro/datasets/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza\"\n",
    "experiment_name = \"ana-flavia-HSD-NCxHSD-NR\"\n",
    "manifest_file = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-HSD-NCxHSD-NR.csv\"\n",
    "metadata_file = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-HSD-NCxHSD-NR.tsv\"\n",
    "overlap = 12\n",
    "phred = 20\n",
    "replace_files = False\n",
    "threads = 6\n",
    "top_n = 20\n",
    "trim = {\"forward_primer\": \"CCTACGGGRSGCAGCAG\", \"overlap\": 8, \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\"}\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8b129f9d",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-08-18T15:40:31.070885Z",
     "start_time": "2023-08-18T15:40:30.611671Z"
    },
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.624787Z",
     "iopub.status.busy": "2024-09-17T20:19:51.624473Z",
     "iopub.status.idle": "2024-09-17T20:19:51.627633Z",
     "shell.execute_reply": "2024-09-17T20:19:51.627217Z"
    },
    "papermill": {
     "duration": 0.006136,
     "end_time": "2024-09-17T20:19:51.628504",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.622368",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "out_dir = os.path.join(base_dir, 'experiments', experiment_name, 'qiime-artifacts')\n",
    "img_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name, 'imgs'))\n",
    "data_dir =  os.path.abspath(os.path.join(base_dir, 'data'))\n",
    "raw_data_dir =  os.path.abspath(os.path.join(data_dir, 'raw'))\n",
    "interim_data_dir =  os.path.abspath(os.path.join(data_dir, 'interim'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f12bd39e",
   "metadata": {
    "papermill": {
     "duration": 0.001641,
     "end_time": "2024-09-17T20:19:51.631804",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.630163",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names and paths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8b15f5fa",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T18:50:41.632125Z",
     "start_time": "2022-06-28T18:50:41.619710Z"
    },
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.635820Z",
     "iopub.status.busy": "2024-09-17T20:19:51.635581Z",
     "iopub.status.idle": "2024-09-17T20:19:51.638516Z",
     "shell.execute_reply": "2024-09-17T20:19:51.638216Z"
    },
    "papermill": {
     "duration": 0.00593,
     "end_time": "2024-09-17T20:19:51.639393",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.633463",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "New qiime-artifacts folder path created: /mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-HSD-NCxHSD-NR/qiime-artifacts\n",
      "New img folder path created: /mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-HSD-NCxHSD-NR/imgs\n"
     ]
    }
   ],
   "source": [
    "# Create path if it not exists\n",
    "if not os.path.isdir(out_dir):\n",
    "    os.makedirs(out_dir)\n",
    "    print(f'New qiime-artifacts folder path created: {out_dir}')\n",
    "\n",
    "# Create path if it not exists\n",
    "if not os.path.isdir(img_folder):\n",
    "    os.makedirs(img_folder)\n",
    "    print(f'New img folder path created: {img_folder}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bd2d3a47",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.643521Z",
     "iopub.status.busy": "2024-09-17T20:19:51.643281Z",
     "iopub.status.idle": "2024-09-17T20:19:51.654601Z",
     "shell.execute_reply": "2024-09-17T20:19:51.654175Z"
    },
    "papermill": {
     "duration": 0.014408,
     "end_time": "2024-09-17T20:19:51.655496",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.641088",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "manifest_df = pd.read_csv(manifest_file)\n",
    "n_directions = len(manifest_df['direction'].unique())\n",
    "if n_directions == 1:\n",
    "    d_type = 'SampleData[SequencesWithQuality]'\n",
    "    v_type = 'SingleEndFastqManifestPhred33'\n",
    "    direction = 'single'\n",
    "elif n_directions == 2:\n",
    "    d_type = 'SampleData[PairedEndSequencesWithQuality]'\n",
    "    v_type = 'PairedEndFastqManifestPhred33'\n",
    "    direction = 'paired'\n",
    "else:\n",
    "    print(f'ERROR: invalid number of directions {n_directions}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1163f4a3",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.659722Z",
     "iopub.status.busy": "2024-09-17T20:19:51.659441Z",
     "iopub.status.idle": "2024-09-17T20:19:51.662263Z",
     "shell.execute_reply": "2024-09-17T20:19:51.661845Z"
    },
    "papermill": {
     "duration": 0.005952,
     "end_time": "2024-09-17T20:19:51.663181",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.657229",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "sufix = str(direction)\n",
    "if trim and len(trim.keys()) == 3:\n",
    "    sufix += f'-{trim}'\n",
    "    \n",
    "# Define the output artifact full path\n",
    "if demux_file is None:\n",
    "    demux_file = os.path.join(out_dir, f'demux-{direction}.qza')\n",
    "demux_view = os.path.join(out_dir, f'demux-{direction}.qzv')\n",
    "demux_file_trim = os.path.join(out_dir, f'demux-{direction}-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, f'demux-{direction}-trim.qzv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d16cfa4d",
   "metadata": {
    "papermill": {
     "duration": 0.001733,
     "end_time": "2024-09-17T20:19:51.666651",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.664918",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "This Step import all `fastq` files in a **QIIME2 Artifact** object and save it to a new `qza` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9cfe1a05",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.670878Z",
     "iopub.status.busy": "2024-09-17T20:19:51.670608Z",
     "iopub.status.idle": "2024-09-17T20:19:51.679460Z",
     "shell.execute_reply": "2024-09-17T20:19:51.679144Z"
    },
    "papermill": {
     "duration": 0.011934,
     "end_time": "2024-09-17T20:19:51.680340",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.668406",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If the metadata file is not defined, use the default metadata file\n",
    "metadata_qa = Metadata.load(metadata_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "611821a3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:44:58.640338Z",
     "start_time": "2022-06-28T18:50:41.633657Z"
    },
    "execution": {
     "iopub.execute_input": "2024-09-17T20:19:51.684565Z",
     "iopub.status.busy": "2024-09-17T20:19:51.684336Z",
     "iopub.status.idle": "2024-09-17T20:20:32.232115Z",
     "shell.execute_reply": "2024-09-17T20:20:32.231625Z"
    },
    "papermill": {
     "duration": 40.551159,
     "end_time": "2024-09-17T20:20:32.233276",
     "exception": false,
     "start_time": "2024-09-17T20:19:51.682117",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-2023.7.1/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-2023.7.1/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    }
   ],
   "source": [
    "# If the demux file does not exist or if the old file will be replaced\n",
    "if not os.path.isfile(demux_file) or replace_files:\n",
    "    \n",
    "    interim_demux_path = os.path.join(interim_data_dir, f'demux-{direction}.qza')\n",
    "    \n",
    "    # Create new file path for interim dir\n",
    "    if not os.path.isdir(interim_data_dir):\n",
    "        !mkdir -p {interim_data_dir}\n",
    "        \n",
    "    # Load Artifact from interim folder\n",
    "    if os.path.isfile(interim_demux_path):\n",
    "        artifact = Artifact.load(interim_demux_path)\n",
    "        \n",
    "    # Create new Artifact using Manifest\n",
    "    else:\n",
    "        artifact = Artifact.import_data(d_type, manifest_file, view_type=v_type)\n",
    "        if not os.path.isfile(interim_demux_path):\n",
    "            artifact.save(interim_demux_path)\n",
    "            \n",
    "    # Filter and Save Artifact\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    artifact.save(demux_file)\n",
    "    \n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)\n",
    "        \n",
    "elif os.path.isfile(demux_file):\n",
    "    \n",
    "    # Load Artifact\n",
    "    artifact = Artifact.load(demux_file)\n",
    "    artifact = filter_samples(demux=artifact, metadata=metadata_qa).filtered_demux\n",
    "    \n",
    "    # Visualization\n",
    "    if os.path.isfile(demux_view):\n",
    "        demux_view_obj = Visualization.load(demux_view)\n",
    "    else:\n",
    "        demux_view_obj = demux.visualizers.summarize(artifact).visualization\n",
    "        Visualization.save(demux_view_obj, filepath=demux_view)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bfa050bf",
   "metadata": {
    "papermill": {
     "duration": 0.001879,
     "end_time": "2024-09-17T20:20:32.237274",
     "exception": false,
     "start_time": "2024-09-17T20:20:32.235395",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step report\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0a2181a7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-28T19:47:56.704841Z",
     "start_time": "2022-06-28T19:47:56.611846Z"
    },
    "execution": {
     "iopub.execute_input": "2024-09-17T20:20:32.241823Z",
     "iopub.status.busy": "2024-09-17T20:20:32.241672Z",
     "iopub.status.idle": "2024-09-17T20:20:32.246920Z",
     "shell.execute_reply": "2024-09-17T20:20:32.246609Z"
    },
    "papermill": {
     "duration": 0.008676,
     "end_time": "2024-09-17T20:20:32.247818",
     "exception": false,
     "start_time": "2024-09-17T20:20:32.239142",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<visualization: Visualization uuid: f79940ab-98bd-4d59-8a5a-ee363cc34a21>\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter serverextension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/lauro/data/f79940ab-98bd-4d59-8a5a-ee363cc34a21')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f79940ab-98bd-4d59-8a5a-ee363cc34a21>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(demux_view_obj)\n",
    "\n",
    "# Render Visualization\n",
    "demux_view_obj"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fa8d370",
   "metadata": {
    "papermill": {
     "duration": 0.001957,
     "end_time": "2024-09-17T20:20:32.251775",
     "exception": false,
     "start_time": "2024-09-17T20:20:32.249818",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "[cutadapt](https://docs.qiime2.org/2022.2/plugins/available/cutadapt/trim-paired/)\n",
    "\n",
    "We trim the forward primer and the reverse complement of the reverse primer from the forward reads. We trim the reverse primer and reverse complement of the forward primer from the reverse reads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d769a304",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-09-17T20:20:32.256365Z",
     "iopub.status.busy": "2024-09-17T20:20:32.256242Z",
     "iopub.status.idle": "2024-09-17T20:20:56.488599Z",
     "shell.execute_reply": "2024-09-17T20:20:56.488101Z"
    },
    "papermill": {
     "duration": 24.236135,
     "end_time": "2024-09-17T20:20:56.489906",
     "exception": false,
     "start_time": "2024-09-17T20:20:32.253771",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application. This may print messages to stdout and/or stderr.\n",
      "The commands to be run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121698_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121698_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121698_50_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121698_51_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121698_50_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121698_51_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121698_50_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121698_51_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.146 s (10.221 µs/read; 5.87 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             14,319\n",
      "  Read 1 with adapter:                  13,866 (96.8%)\n",
      "  Read 2 with adapter:                  12,833 (89.6%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,875 (13.1%)\n",
      "Pairs written (passing filters):        12,444 (86.9%)\n",
      "\n",
      "Total basepairs processed:     7,302,690 bp\n",
      "  Read 1:     4,367,295 bp\n",
      "  Read 2:     2,935,395 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      5,801,125 bp (79.4%)\n",
      "  Read 1:     3,540,895 bp\n",
      "  Read 2:     2,260,230 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 13866 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t5\t0.0\t0\t5\n",
      "18\t2170\t0.0\t0\t2170\n",
      "19\t2437\t0.0\t0\t2437\n",
      "20\t2494\t0.0\t0\t2494\n",
      "21\t2324\t0.0\t0\t2324\n",
      "22\t2426\t0.0\t0\t2426\n",
      "23\t2006\t0.0\t0\t2006\n",
      "24\t1\t0.0\t0\t1\n",
      "45\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 12833 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t3\t0.0\t0\t3\n",
      "21\t2351\t0.0\t0\t2351\n",
      "22\t2314\t0.0\t0\t2314\n",
      "23\t2129\t0.0\t0\t2129\n",
      "24\t2175\t0.0\t0\t2175\n",
      "25\t2020\t0.0\t0\t2020\n",
      "26\t1841\t0.0\t0\t1841\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121699_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121699_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121699_52_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121699_53_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121699_52_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121699_53_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121699_52_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121699_53_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.199 s (9.505 µs/read; 6.31 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,959\n",
      "  Read 1 with adapter:                  20,357 (97.1%)\n",
      "  Read 2 with adapter:                  19,037 (90.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,450 (11.7%)\n",
      "Pairs written (passing filters):        18,509 (88.3%)\n",
      "\n",
      "Total basepairs processed:    10,689,090 bp\n",
      "  Read 1:     6,392,495 bp\n",
      "  Read 2:     4,296,595 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      8,628,512 bp (80.7%)\n",
      "  Read 1:     5,266,608 bp\n",
      "  Read 2:     3,361,904 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 20357 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3085\t0.0\t0\t3085\n",
      "19\t3612\t0.0\t0\t3612\n",
      "20\t3759\t0.0\t0\t3759\n",
      "21\t3463\t0.0\t0\t3463\n",
      "22\t3487\t0.0\t0\t3487\n",
      "23\t2937\t0.0\t0\t2937\n",
      "24\t5\t0.0\t0\t5\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 19037 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t6\t0.0\t0\t6\n",
      "21\t3547\t0.0\t0\t3547\n",
      "22\t3449\t0.0\t0\t3449\n",
      "23\t3103\t0.0\t0\t3103\n",
      "24\t3209\t0.0\t0\t3209\n",
      "25\t2928\t0.0\t0\t2928\n",
      "26\t2792\t0.0\t0\t2792\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121700_54_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121700_55_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121700_54_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121700_55_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121700_54_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121700_55_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121700_54_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121700_55_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.172 s (8.540 µs/read; 7.03 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             20,194\n",
      "  Read 1 with adapter:                  19,610 (97.1%)\n",
      "  Read 2 with adapter:                  18,045 (89.4%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            2,655 (13.1%)\n",
      "Pairs written (passing filters):        17,539 (86.9%)\n",
      "\n",
      "Total basepairs processed:    10,298,940 bp\n",
      "  Read 1:     6,159,170 bp\n",
      "  Read 2:     4,139,770 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      8,176,314 bp (79.4%)\n",
      "  Read 1:     4,990,682 bp\n",
      "  Read 2:     3,185,632 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 19610 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t3\t0.0\t0\t3\n",
      "17\t7\t0.0\t0\t7\n",
      "18\t3003\t0.0\t0\t3003\n",
      "19\t3491\t0.0\t0\t3491\n",
      "20\t3617\t0.0\t0\t3617\n",
      "21\t3224\t0.0\t0\t3224\n",
      "22\t3416\t0.0\t0\t3416\n",
      "23\t2844\t0.0\t0\t2844\n",
      "24\t4\t0.0\t0\t4\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 18045 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t10\t0.0\t0\t10\n",
      "21\t3352\t0.0\t0\t3352\n",
      "22\t3231\t0.0\t0\t3231\n",
      "23\t2977\t0.0\t0\t2977\n",
      "24\t3007\t0.0\t0\t3007\n",
      "25\t2837\t0.0\t0\t2837\n",
      "26\t2625\t0.0\t0\t2625\n",
      "27\t4\t0.0\t0\t4\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121701_56_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121701_57_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121701_56_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121701_57_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121701_56_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121701_57_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121701_56_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121701_57_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.313 s (4.645 µs/read; 12.92 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             67,293\n",
      "  Read 1 with adapter:                  65,501 (97.3%)\n",
      "  Read 2 with adapter:                  61,930 (92.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            6,987 (10.4%)\n",
      "Pairs written (passing filters):        60,306 (89.6%)\n",
      "\n",
      "Total basepairs processed:    34,319,430 bp\n",
      "  Read 1:    20,524,365 bp\n",
      "  Read 2:    13,795,065 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     28,113,964 bp (81.9%)\n",
      "  Read 1:    17,159,382 bp\n",
      "  Read 2:    10,954,582 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 65501 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t2\t0.0\t0\t2\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t19\t0.0\t0\t19\n",
      "18\t10169\t0.0\t0\t10169\n",
      "19\t11601\t0.0\t0\t11601\n",
      "20\t12117\t0.0\t0\t12117\n",
      "21\t10719\t0.0\t0\t10719\n",
      "22\t11255\t0.0\t0\t11255\n",
      "23\t9603\t0.0\t0\t9603\n",
      "24\t7\t0.0\t0\t7\n",
      "27\t1\t0.0\t0\t1\n",
      "31\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 61930 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "18\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t18\t0.0\t0\t18\n",
      "21\t11697\t0.0\t0\t11697\n",
      "22\t11095\t0.0\t0\t11095\n",
      "23\t10267\t0.0\t0\t10267\n",
      "24\t10352\t0.0\t0\t10352\n",
      "25\t9484\t0.0\t0\t9484\n",
      "26\t9007\t0.0\t0\t9007\n",
      "27\t6\t0.0\t0\t6\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121702_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121702_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121702_58_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121702_59_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121702_58_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121702_59_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121702_58_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121702_59_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n",
      "Finished in 0.124 s (14.920 µs/read; 4.02 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              8,334\n",
      "  Read 1 with adapter:                   8,081 (97.0%)\n",
      "  Read 2 with adapter:                   7,509 (90.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            1,046 (12.6%)\n",
      "Pairs written (passing filters):         7,288 (87.4%)\n",
      "\n",
      "Total basepairs processed:     4,250,340 bp\n",
      "  Read 1:     2,541,870 bp\n",
      "  Read 2:     1,708,470 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      3,397,413 bp (79.9%)\n",
      "  Read 1:     2,073,898 bp\n",
      "  Read 2:     1,323,515 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 8081 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t3\t0.0\t0\t3\n",
      "18\t1267\t0.0\t0\t1267\n",
      "19\t1465\t0.0\t0\t1465\n",
      "20\t1495\t0.0\t0\t1495\n",
      "21\t1281\t0.0\t0\t1281\n",
      "22\t1390\t0.0\t0\t1390\n",
      "23\t1178\t0.0\t0\t1178\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 7509 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "20\t3\t0.0\t0\t3\n",
      "21\t1388\t0.0\t0\t1388\n",
      "22\t1308\t0.0\t0\t1308\n",
      "23\t1248\t0.0\t0\t1248\n",
      "24\t1240\t0.0\t0\t1240\n",
      "25\t1164\t0.0\t0\t1164\n",
      "26\t1154\t0.0\t0\t1154\n",
      "27\t2\t0.0\t0\t2\n",
      "31\t1\t0.0\t0\t1\n",
      "59\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121707_68_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121707_69_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121707_68_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121707_69_L001_R2_001.fastq.gz\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121707_68_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121707_69_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121707_68_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121707_69_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.447 s (5.480 µs/read; 10.95 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             81,485\n",
      "  Read 1 with adapter:                  79,303 (97.3%)\n",
      "  Read 2 with adapter:                  74,804 (91.8%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            8,666 (10.6%)\n",
      "Pairs written (passing filters):        72,819 (89.4%)\n",
      "\n",
      "Total basepairs processed:    41,557,350 bp\n",
      "  Read 1:    24,852,925 bp\n",
      "  Read 2:    16,704,425 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     33,947,823 bp (81.7%)\n",
      "  Read 1:    20,720,493 bp\n",
      "  Read 2:    13,227,330 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 79303 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "9\t2\t0.3\t0\t2\n",
      "10\t2\t0.1\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "14\t4\t0.0\t0\t4\n",
      "15\t3\t0.0\t0\t3\n",
      "16\t2\t0.0\t0\t2\n",
      "17\t34\t0.0\t0\t34\n",
      "18\t12320\t0.0\t0\t12320\n",
      "19\t14302\t0.0\t0\t14302\n",
      "20\t14387\t0.0\t0\t14387\n",
      "21\t13117\t0.0\t0\t13117\n",
      "22\t13446\t0.0\t0\t13446\n",
      "23\t11676\t0.0\t0\t11676\n",
      "24\t5\t0.0\t0\t5\n",
      "28\t1\t0.0\t0\t1\n",
      "30\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 74804 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "10\t2\t0.1\t0\t2\n",
      "14\t1\t0.0\t0\t1\n",
      "15\t1\t0.0\t0\t1\n",
      "17\t1\t0.0\t0\t1\n",
      "19\t2\t0.0\t0\t2\n",
      "20\t25\t0.0\t0\t25\n",
      "21\t14012\t0.0\t0\t14012\n",
      "22\t13493\t0.0\t0\t13493\n",
      "23\t12382\t0.0\t0\t12382\n",
      "24\t12641\t0.0\t0\t12641\n",
      "25\t11252\t0.0\t0\t11252\n",
      "26\t10983\t0.0\t0\t10983\n",
      "27\t8\t0.0\t0\t8\n",
      "53\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121708_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121708_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121708_70_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121708_71_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121708_70_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121708_71_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121708_70_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121708_71_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.200 s (7.339 µs/read; 8.18 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             27,196\n",
      "  Read 1 with adapter:                  26,427 (97.2%)\n",
      "  Read 2 with adapter:                  24,481 (90.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,391 (12.5%)\n",
      "Pairs written (passing filters):        23,805 (87.5%)\n",
      "\n",
      "Total basepairs processed:    13,869,960 bp\n",
      "  Read 1:     8,294,780 bp\n",
      "  Read 2:     5,575,180 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     11,097,905 bp (80.0%)\n",
      "  Read 1:     6,774,112 bp\n",
      "  Read 2:     4,323,793 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 26427 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t2\t0.0\t0\t2\n",
      "17\t6\t0.0\t0\t6\n",
      "18\t4153\t0.0\t0\t4153\n",
      "19\t4745\t0.0\t0\t4745\n",
      "20\t4859\t0.0\t0\t4859\n",
      "21\t4346\t0.0\t0\t4346\n",
      "22\t4473\t0.0\t0\t4473\n",
      "23\t3839\t0.0\t0\t3839\n",
      "24\t3\t0.0\t0\t3\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 24481 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t1\t0.0\t0\t1\n",
      "20\t5\t0.0\t0\t5\n",
      "21\t4531\t0.0\t0\t4531\n",
      "22\t4333\t0.0\t0\t4333\n",
      "23\t4146\t0.0\t0\t4146\n",
      "24\t4150\t0.0\t0\t4150\n",
      "25\t3714\t0.0\t0\t3714\n",
      "26\t3599\t0.0\t0\t3599\n",
      "27\t1\t0.0\t0\t1\n",
      "28\t1\t0.0\t0\t1\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121709_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121709_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121709_72_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121709_73_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121709_72_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121709_73_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121709_72_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121709_73_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finished in 0.215 s (6.553 µs/read; 9.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:             32,857\n",
      "  Read 1 with adapter:                  31,953 (97.2%)\n",
      "  Read 2 with adapter:                  29,876 (90.9%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:            3,780 (11.5%)\n",
      "Pairs written (passing filters):        29,077 (88.5%)\n",
      "\n",
      "Total basepairs processed:    16,757,070 bp\n",
      "  Read 1:    10,021,385 bp\n",
      "  Read 2:     6,735,685 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):     13,555,322 bp (80.9%)\n",
      "  Read 1:     8,273,880 bp\n",
      "  Read 2:     5,281,442 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 31953 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t2\t0.0\t0\t2\n",
      "12\t1\t0.0\t0\t1\n",
      "13\t1\t0.0\t0\t1\n",
      "14\t1\t0.0\t0\t1\n",
      "16\t1\t0.0\t0\t1\n",
      "17\t9\t0.0\t0\t9\n",
      "18\t4981\t0.0\t0\t4981\n",
      "19\t5717\t0.0\t0\t5717\n",
      "20\t5839\t0.0\t0\t5839\n",
      "21\t5216\t0.0\t0\t5216\n",
      "22\t5501\t0.0\t0\t5501\n",
      "23\t4678\t0.0\t0\t4678\n",
      "24\t5\t0.0\t0\t5\n",
      "29\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 29876 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "11\t1\t0.0\t0\t1\n",
      "15\t2\t0.0\t0\t2\n",
      "19\t1\t0.0\t0\t1\n",
      "20\t12\t0.0\t0\t12\n",
      "21\t5470\t0.0\t0\t5470\n",
      "22\t5373\t0.0\t0\t5373\n",
      "23\t5010\t0.0\t0\t5010\n",
      "24\t5115\t0.0\t0\t5115\n",
      "25\t4536\t0.0\t0\t4536\n",
      "26\t4354\t0.0\t0\t4354\n",
      "27\t2\t0.0\t0\t2\n",
      "\n",
      "Command: cutadapt --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121710_74_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121710_75_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121710_74_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121710_75_L001_R2_001.fastq.gz\n",
      "\n",
      "This is cutadapt 4.4 with Python 3.8.15\n",
      "Command line parameters: --cores 6 --error-rate 0.01 --times 1 --overlap 8 --minimum-length 1 -q 0,0 --quality-base 33 -o /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121710_74_L001_R1_001.fastq.gz -p /tmp/q2-CasavaOneEightSingleLanePerSampleDirFmt-xd6mu4sm/S210421121710_75_L001_R2_001.fastq.gz --front CCTACGGGRSGCAGCAG -G GGACTACHVGGGTWTCTAAT --no-indels --match-read-wildcards --discard-untrimmed /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121710_74_L001_R1_001.fastq.gz /tmp/qiime2/lauro/data/52e2b061-359f-40a7-8b67-0387537111da/data/S210421121710_75_L001_R2_001.fastq.gz\n",
      "Processing paired-end reads on 6 cores ...\n",
      "Finished in 0.096 s (19.561 µs/read; 3.07 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total read pairs processed:              4,908\n",
      "  Read 1 with adapter:                   4,754 (96.9%)\n",
      "  Read 2 with adapter:                   4,257 (86.7%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Pairs that were too short:                   0 (0.0%)\n",
      "Pairs discarded as untrimmed:              773 (15.7%)\n",
      "Pairs written (passing filters):         4,135 (84.3%)\n",
      "\n",
      "Total basepairs processed:     2,503,080 bp\n",
      "  Read 1:     1,496,940 bp\n",
      "  Read 2:     1,006,140 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "  Read 1:             0 bp\n",
      "  Read 2:             0 bp\n",
      "Total written (filtered):      1,927,695 bp (77.0%)\n",
      "  Read 1:     1,176,557 bp\n",
      "  Read 2:       751,138 bp\n",
      "\n",
      "=== First read: Adapter 1 ===\n",
      "\n",
      "Sequence: CCTACGGGRSGCAGCAG; Type: regular 5'; Length: 17; Trimmed: 4754 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-17 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "18\t757\t0.0\t0\t757\n",
      "19\t836\t0.0\t0\t836\n",
      "20\t829\t0.0\t0\t829\n",
      "21\t772\t0.0\t0\t772\n",
      "22\t824\t0.0\t0\t824\n",
      "23\t735\t0.0\t0\t735\n",
      "24\t1\t0.0\t0\t1\n",
      "\n",
      "\n",
      "=== Second read: Adapter 2 ===\n",
      "\n",
      "Sequence: GGACTACHVGGGTWTCTAAT; Type: regular 5'; Length: 20; Trimmed: 4257 times\n",
      "\n",
      "Minimum overlap: 8\n",
      "No. of allowed errors:\n",
      "1-20 bp: 0\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "15\t1\t0.0\t0\t1\n",
      "20\t2\t0.0\t0\t2\n",
      "21\t819\t0.0\t0\t819\n",
      "22\t759\t0.0\t0\t759\n",
      "23\t673\t0.0\t0\t673\n",
      "24\t752\t0.0\t0\t752\n",
      "25\t638\t0.0\t0\t638\n",
      "26\t613\t0.0\t0\t613\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-2023.7.1/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-2023.7.1/lib/python3.8/site-packages/q2_demux/_summarize/_visualizer.py:191: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  context['result_data'] = context['result_data'].append(df)\n"
     ]
    }
   ],
   "source": [
    "demux_file_trim = os.path.join(out_dir, f'demux-{direction}-trim.qza')\n",
    "demux_view_trim = os.path.join(out_dir, f'demux-{direction}-trim.qzv')\n",
    "\n",
    "if trim and (not os.path.isfile(demux_file_trim) or replace_files):\n",
    "    forward_primer = [trim['forward_primer']] # ['CCTACGGGRSGCAGCAG']\n",
    "    reverse_primer = [trim['reverse_primer']] # ['GGACTACHVGGGTWTCTAAT']\n",
    "    forward_reverse_complement = [str(Seq(forward_primer[0]).reverse_complement())]\n",
    "    reverse_reverse_complement = [str(Seq(reverse_primer[0]).reverse_complement())]\n",
    "\n",
    "    res = None\n",
    "    \n",
    "    if n_directions == 1:\n",
    "        res = trim_single(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front=forward_primer,\n",
    "            adapter=reverse_reverse_complement,\n",
    "            cores=threads,\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    elif n_directions == 2:\n",
    "        res = trim_paired(\n",
    "            demultiplexed_sequences=artifact,\n",
    "            front_f=forward_primer,\n",
    "            front_r=reverse_primer,\n",
    "            # adapter_f=reverse_reverse_complement,\n",
    "            # adapter_r=forward_reverse_complement,\n",
    "            cores=threads,\n",
    "            overlap=trim['overlap'],\n",
    "            indels=False,\n",
    "            match_read_wildcards=True,\n",
    "            match_adapter_wildcards=True,\n",
    "            error_rate=0.01,\n",
    "            discard_untrimmed=True,\n",
    "        ).trimmed_sequences\n",
    "    else:\n",
    "        print(f'ERROR: invalid number of directions {n_directions}')\n",
    "\n",
    "    if res is not None:\n",
    "        res.save(demux_file_trim)\n",
    "        Visualization.save(demux.visualizers.summarize(res).visualization, filepath=demux_view_trim)"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.15"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 71.460409,
   "end_time": "2024-09-17T20:20:57.409525",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-prepare-data.ipynb",
   "output_path": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-HSD-NCxHSD-NR-trim/nb-executed-steps/step-prepare-data-ana-flavia-HSD-NCxHSD-NR-trim.ipynb",
   "parameters": {
    "base_dir": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/mnt/nupeb/rede-micro/datasets/16S_classifiers_qiime2/silva-138-99-nb-classifier.qza",
    "experiment_name": "ana-flavia-HSD-NCxHSD-NR",
    "manifest_file": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/manifest/manifest-ana-flavia-HSD-NCxHSD-NR.csv",
    "metadata_file": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-HSD-NCxHSD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2024-09-17T20:19:45.949116",
   "version": "2.6.0"
  },
  "toc-autonumbering": false,
  "toc-showcode": false,
  "toc-showmarkdowntxt": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}