{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0e7f2330",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-22T19:36:33.828188Z",
     "start_time": "2022-09-22T19:36:33.817323Z"
    },
    "papermill": {
     "duration": 0.01218,
     "end_time": "2024-10-07T22:44:18.991483",
     "exception": false,
     "start_time": "2024-10-07T22:44:18.979303",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### STEP : Diversity Analysis\n",
    "\n",
    "\n",
    "\n",
    "#### Example\n",
    "\n",
    "- [“Moving Pictures” tutorial](https://docs.qiime2.org/2022.8/tutorials/moving-pictures-usage/?highlight=ancom)\n",
    "- [Biostats ANCOM](http://scikit-bio.org/docs/0.4.2/generated/generated/skbio.stats.composition.ancom.html)\n",
    "\n",
    "\n",
    "#### Methods\n",
    "- [composition](https://docs.qiime2.org/2022.8/plugins/available/composition/)\n",
    "- [composition add_pseudocount](https://docs.qiime2.org/2022.8/plugins/available/composition/add-pseudocount/): Increment all counts in table by pseudocount.\n",
    "- [composition ancom](https://docs.qiime2.org/2022.8/plugins/available/composition/ancom/): Apply ANCOM to identify features that differ in abundance."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "843debbf",
   "metadata": {
    "papermill": {
     "duration": 0.009035,
     "end_time": "2024-10-07T22:44:19.011858",
     "exception": false,
     "start_time": "2024-10-07T22:44:19.002823",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "965e28e5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:21:12.095385Z",
     "start_time": "2024-07-09T15:21:12.087592Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:19.023657Z",
     "iopub.status.busy": "2024-10-07T22:44:19.023347Z",
     "iopub.status.idle": "2024-10-07T22:44:23.932999Z",
     "shell.execute_reply": "2024-10-07T22:44:23.932501Z"
    },
    "papermill": {
     "duration": 4.916402,
     "end_time": "2024-10-07T22:44:23.934209",
     "exception": false,
     "start_time": "2024-10-07T22:44:19.017807",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import pandas as pd\n",
    "from qiime2 import Artifact\n",
    "from qiime2 import Visualization\n",
    "from qiime2 import Metadata\n",
    "\n",
    "from qiime2.plugins.composition.visualizers import ancom\n",
    "from qiime2.plugins.composition.methods import add_pseudocount\n",
    "\n",
    "from qiime2.plugins.feature_table.methods import filter_samples\n",
    "from qiime2.plugins.feature_table.methods import filter_seqs\n",
    "from qiime2.plugins.feature_table.methods import relative_frequency\n",
    "\n",
    "from qiime2.plugins.taxa.methods import collapse\n",
    "\n",
    "import biom\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530ceb60",
   "metadata": {
    "papermill": {
     "duration": 0.001769,
     "end_time": "2024-10-07T22:44:23.937985",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.936216",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18d874ea",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:21:12.126684Z",
     "start_time": "2024-07-09T15:21:12.123749Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:23.942190Z",
     "iopub.status.busy": "2024-10-07T22:44:23.941946Z",
     "iopub.status.idle": "2024-10-07T22:44:23.944605Z",
     "shell.execute_reply": "2024-10-07T22:44:23.944266Z"
    },
    "papermill": {
     "duration": 0.005759,
     "end_time": "2024-10-07T22:44:23.945484",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.939725",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "metadata_file = '/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/metadata/metadata-ana-flavia-NCxNR.tsv'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'rede-micro', 'redemicro-miliane-nutri')\n",
    "base_dir = '/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri'\n",
    "experiment_name = 'miliane-CxAC-trim'\n",
    "class_col = 'group-id'\n",
    "replace_files = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9c4153fc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:23.949495Z",
     "iopub.status.busy": "2024-10-07T22:44:23.949373Z",
     "iopub.status.idle": "2024-10-07T22:44:23.951971Z",
     "shell.execute_reply": "2024-10-07T22:44:23.951682Z"
    },
    "papermill": {
     "duration": 0.005721,
     "end_time": "2024-10-07T22:44:23.952921",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.947200",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "base_dir = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/mnt/nupeb/rede-micro/datasets/16S_classifiers_qiime2/silva138.1_AB_V3-V4_classifier.qza\"\n",
    "experiment_name = \"ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR\"\n",
    "manifest_file = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/manifest_STD-NCxSTD-NRxHSD-NCxHSD-NR.csv\"\n",
    "metadata_file = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/metadata_STD-NCxSTD-NRxHSD-NCxHSD-NR.tsv\"\n",
    "overlap = 12\n",
    "phred = 20\n",
    "replace_files = False\n",
    "threads = 6\n",
    "top_n = 20\n",
    "trim = {\"forward_primer\": \"CCTACGGGRSGCAGCAG\", \"overlap\": 8, \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\"}\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f38617d9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:21:12.158084Z",
     "start_time": "2024-07-09T15:21:12.152310Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:23.957002Z",
     "iopub.status.busy": "2024-10-07T22:44:23.956881Z",
     "iopub.status.idle": "2024-10-07T22:44:23.959216Z",
     "shell.execute_reply": "2024-10-07T22:44:23.958872Z"
    },
    "papermill": {
     "duration": 0.005448,
     "end_time": "2024-10-07T22:44:23.960113",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.954665",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "experiment_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name))\n",
    "img_folder = os.path.abspath(os.path.join(experiment_folder, 'imgs'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea9b90fb",
   "metadata": {
    "papermill": {
     "duration": 0.001742,
     "end_time": "2024-10-07T22:44:23.963646",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.961904",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names, paths and flags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "45727eef",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:23:01.978859Z",
     "start_time": "2024-07-09T15:23:01.973238Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:23.967794Z",
     "iopub.status.busy": "2024-10-07T22:44:23.967609Z",
     "iopub.status.idle": "2024-10-07T22:44:23.969754Z",
     "shell.execute_reply": "2024-10-07T22:44:23.969352Z"
    },
    "papermill": {
     "duration": 0.005256,
     "end_time": "2024-10-07T22:44:23.970672",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.965416",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# QIIME2 Artifacts folder\n",
    "qiime_folder = os.path.join(experiment_folder, 'qiime-artifacts')\n",
    "\n",
    "# Input - DADA2 Artifacts\n",
    "dada2_tabs_path = os.path.join(qiime_folder, 'dada2-tabs.qza')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c3719207dcdfebfd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:23:10.616148Z",
     "start_time": "2024-07-09T15:23:10.495059Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:23.974986Z",
     "iopub.status.busy": "2024-10-07T22:44:23.974766Z",
     "iopub.status.idle": "2024-10-07T22:44:24.091150Z",
     "shell.execute_reply": "2024-10-07T22:44:24.088549Z"
    },
    "papermill": {
     "duration": 0.123759,
     "end_time": "2024-10-07T22:44:24.096234",
     "exception": false,
     "start_time": "2024-10-07T22:44:23.972475",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/lauro/rede-micro/redemicro-ana-flavia-nutri\r\n"
     ]
    }
   ],
   "source": [
    "qiime_folder\n",
    "!pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0999a1b2",
   "metadata": {
    "papermill": {
     "duration": 0.011088,
     "end_time": "2024-10-07T22:44:24.119181",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.108093",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "### Load input files\n",
    "\n",
    "This Step import the QIIME2 `FeatureTable[Frequency]` Artifact and the `Metadata` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2e59f012aa21bed8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:21:28.257817Z",
     "start_time": "2024-07-09T15:21:28.135390Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:24.136257Z",
     "iopub.status.busy": "2024-10-07T22:44:24.135929Z",
     "iopub.status.idle": "2024-10-07T22:44:24.256333Z",
     "shell.execute_reply": "2024-10-07T22:44:24.253782Z"
    },
    "papermill": {
     "duration": 0.133001,
     "end_time": "2024-10-07T22:44:24.261723",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.128722",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access '../data/raw/metadata': No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "metadata_file\n",
    "!ls ../data/raw/metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8f299d1a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-07-09T15:23:19.592703Z",
     "start_time": "2024-07-09T15:23:19.555512Z"
    },
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:24.288791Z",
     "iopub.status.busy": "2024-10-07T22:44:24.287971Z",
     "iopub.status.idle": "2024-10-07T22:44:24.527446Z",
     "shell.execute_reply": "2024-10-07T22:44:24.527010Z"
    },
    "papermill": {
     "duration": 0.254924,
     "end_time": "2024-10-07T22:44:24.528737",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.273813",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Load Metadata\n",
    "metadata_qa = Metadata.load(metadata_file)\n",
    "\n",
    "#Load FeatureTable[Frequency]\n",
    "tabs = Artifact.load(dada2_tabs_path)\n",
    "\n",
    "# Filter FeatureTable[Frequency | RelativeFrequency | PresenceAbsence | Composition] based on Metadata sample ID values\n",
    "tabs = filter_samples(\n",
    "    table=tabs,\n",
    "    metadata=metadata_qa,\n",
    ").filtered_table"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9a446c4",
   "metadata": {
    "papermill": {
     "duration": 0.00195,
     "end_time": "2024-10-07T22:44:24.532760",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.530810",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# ANCOM\n",
    "\n",
    "Apply Analysis of Composition of Microbiomes (ANCOM) to identify features\n",
    "that are differentially abundant across groups.\n",
    "\n",
    "- [composition add_pseudocount](https://docs.qiime2.org/2022.8/plugins/available/composition/add-pseudocount/): Increment all counts in table by pseudocount.\n",
    "- [composition ancom](https://docs.qiime2.org/2022.8/plugins/available/composition/ancom/): Apply ANCOM to identify features that differ in abundance."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cda81bf5",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:24.537539Z",
     "iopub.status.busy": "2024-10-07T22:44:24.537394Z",
     "iopub.status.idle": "2024-10-07T22:44:24.540767Z",
     "shell.execute_reply": "2024-10-07T22:44:24.540290Z"
    },
    "papermill": {
     "duration": 0.006897,
     "end_time": "2024-10-07T22:44:24.541648",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.534751",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Select class column\n",
    "column = metadata_qa.get_column(class_col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "2f023bcc",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:24.546075Z",
     "iopub.status.busy": "2024-10-07T22:44:24.545953Z",
     "iopub.status.idle": "2024-10-07T22:44:24.786181Z",
     "shell.execute_reply": "2024-10-07T22:44:24.785610Z"
    },
    "papermill": {
     "duration": 0.243851,
     "end_time": "2024-10-07T22:44:24.787380",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.543529",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Create the pseudocount table\n",
    "composition_tab = add_pseudocount(table = tabs).composition_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "2999c2f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-10-07T22:44:24.792565Z",
     "iopub.status.busy": "2024-10-07T22:44:24.792378Z",
     "iopub.status.idle": "2024-10-07T23:51:39.132066Z",
     "shell.execute_reply": "2024-10-07T23:51:39.131487Z"
    },
    "papermill": {
     "duration": 4034.343504,
     "end_time": "2024-10-07T23:51:39.133258",
     "exception": false,
     "start_time": "2024-10-07T22:44:24.789754",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating ANCOM with: sqrt mean_difference\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/scipy/stats/_axis_nan_policy.py:531: ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite\n",
      "  res = hypotest_fun_out(*samples, **kwds)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR: Calculating ANCOM with: sqrt mean_difference\n",
      "<lambda>() takes 2 positional arguments but 4 were given\n",
      "Calculating ANCOM with: sqrt f_statistic\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/scipy/stats/_axis_nan_policy.py:531: ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite\n",
      "  res = hypotest_fun_out(*samples, **kwds)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/q2_composition/_ancom.py:103: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating ANCOM with: log mean_difference\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/scipy/stats/_axis_nan_policy.py:531: ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite\n",
      "  res = hypotest_fun_out(*samples, **kwds)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR: Calculating ANCOM with: log mean_difference\n",
      "<lambda>() takes 2 positional arguments but 4 were given\n",
      "Calculating ANCOM with: log f_statistic\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/scipy/stats/_axis_nan_policy.py:531: ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite\n",
      "  res = hypotest_fun_out(*samples, **kwds)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/q2_composition/_ancom.py:103: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating ANCOM with: clr mean_difference\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/scipy/stats/_axis_nan_policy.py:531: ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite\n",
      "  res = hypotest_fun_out(*samples, **kwds)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR: Calculating ANCOM with: clr mean_difference\n",
      "<lambda>() takes 2 positional arguments but 4 were given\n",
      "Calculating ANCOM with: clr f_statistic\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/scipy/stats/_axis_nan_policy.py:531: ConstantInputWarning: Each of the input arrays is constant; the F statistic is not defined or infinite\n",
      "  res = hypotest_fun_out(*samples, **kwds)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/q2_composition/_ancom.py:103: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n"
     ]
    }
   ],
   "source": [
    "# Create visualizations for each parameter combination for the ANCOM  method.\n",
    "transform_functions = ('sqrt', 'log', 'clr')\n",
    "difference_functions = ('mean_difference', 'f_statistic')\n",
    "for t in transform_functions:\n",
    "    for d in difference_functions:\n",
    "        print(f\"Calculating ANCOM with: {t} {d}\")\n",
    "        try:\n",
    "            ancom_viz = ancom(\n",
    "                table = composition_tab, \n",
    "                metadata = column, \n",
    "                transform_function = t, \n",
    "                difference_function = d\n",
    "            ).visualization\n",
    "            view_name = os.path.join(qiime_folder, f'ancom-{t}-{d}.qzv')\n",
    "            ancom_viz.save(view_name)\n",
    "        except Exception as e:\n",
    "            print(f\"ERROR: Calculating ANCOM with: {t} {d}\")\n",
    "            print(e)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2620aa4515e5d784",
   "metadata": {
    "papermill": {
     "duration": 0.002543,
     "end_time": "2024-10-07T23:51:39.138444",
     "exception": false,
     "start_time": "2024-10-07T23:51:39.135901",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "# LEfSe\n",
    "\n",
    "LEfSe (Linear discriminant analysis Effect Size) determines the features (organisms, clades, operational taxonomic units, genes, or functions) most likely to explain differences between classes by coupling standard tests for statistical significance with additional tests encoding biological consistency and effect relevance.\n",
    "\n",
    "- https://huttenhower.sph.harvard.edu/lefse/\n",
    "- https://github.com/statonlab/BiGG2020_CrackNAg/wiki/qiime2-to-lefse\n",
    "- https://github.com/biobakery/biobakery/wiki/lefse#2-lefse--conda-docker-vm-"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b02ffe174ccdde70",
   "metadata": {
    "papermill": {
     "duration": 0.002335,
     "end_time": "2024-10-07T23:51:39.143256",
     "exception": false,
     "start_time": "2024-10-07T23:51:39.140921",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 4041.60162,
   "end_time": "2024-10-07T23:51:39.769930",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-abundance-analysis.ipynb",
   "output_path": "/home/lauro/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR/nb-executed-steps/step-abundance-analysis-ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR.ipynb",
   "parameters": {
    "base_dir": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/mnt/nupeb/rede-micro/datasets/16S_classifiers_qiime2/silva138.1_AB_V3-V4_classifier.qza",
    "experiment_name": "ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR",
    "manifest_file": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/manifest_STD-NCxSTD-NRxHSD-NCxHSD-NR.csv",
    "metadata_file": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/metadata_STD-NCxSTD-NRxHSD-NCxHSD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2024-10-07T22:44:18.168310",
   "version": "2.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}