{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0e7f2330",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-09-22T19:36:33.828188Z",
     "start_time": "2022-09-22T19:36:33.817323Z"
    },
    "papermill": {
     "duration": 0.014516,
     "end_time": "2024-11-04T20:09:30.186929",
     "exception": false,
     "start_time": "2024-11-04T20:09:30.172413",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### STEP : PICRUST2 Analysis\n",
    "\n",
    "\n",
    "\n",
    "#### Example\n",
    "\n",
    "- [PICRUST2 tutorial](https://github.com/picrust/picrust2/wiki/q2-picrust2-Tutorial)\n",
    "- [Limitations](https://github.com/picrust/picrust2/wiki/Key-Limitations)\n",
    "\n",
    "\n",
    "#### Methods\n",
    "- [composition](https://docs.qiime2.org/2022.8/plugins/available/composition/)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "843debbf",
   "metadata": {
    "papermill": {
     "duration": 0.008889,
     "end_time": "2024-11-04T20:09:30.209201",
     "exception": false,
     "start_time": "2024-11-04T20:09:30.200312",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Setup and settings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "965e28e5",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.478806Z",
     "start_time": "2022-11-09T19:20:40.457657Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:30.221044Z",
     "iopub.status.busy": "2024-11-04T20:09:30.220773Z",
     "iopub.status.idle": "2024-11-04T20:09:35.207862Z",
     "shell.execute_reply": "2024-11-04T20:09:35.207349Z"
    },
    "papermill": {
     "duration": 4.993976,
     "end_time": "2024-11-04T20:09:35.209100",
     "exception": false,
     "start_time": "2024-11-04T20:09:30.215124",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Importing packages\n",
    "import os\n",
    "import biom\n",
    "import pandas as pd\n",
    "from qiime2 import Artifact\n",
    "from qiime2 import Visualization\n",
    "from qiime2 import Metadata\n",
    "\n",
    "from qiime2.plugins.feature_table.visualizers import summarize\n",
    "from qiime2.plugins.diversity.pipelines import core_metrics\n",
    "\n",
    "from picrust2.pipeline import full_pipeline\n",
    "from picrust2.default import (default_ref_dir, default_tables, default_regroup_map, default_pathway_map)\n",
    "from qiime2.plugins import picrust2\n",
    "from qiime2.plugins.feature_table.methods import filter_samples\n",
    "from qiime2.plugins.feature_table.methods import filter_seqs\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530ceb60",
   "metadata": {
    "papermill": {
     "duration": 0.002154,
     "end_time": "2024-11-04T20:09:35.213525",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.211371",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Receiving the parameters\n",
    "\n",
    "The following cell can receive parameters using the [papermill](https://papermill.readthedocs.io/en/latest/) tool."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "18d874ea",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.483140Z",
     "start_time": "2022-11-09T19:20:47.480386Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.218370Z",
     "iopub.status.busy": "2024-11-04T20:09:35.218128Z",
     "iopub.status.idle": "2024-11-04T20:09:35.220903Z",
     "shell.execute_reply": "2024-11-04T20:09:35.220457Z"
    },
    "papermill": {
     "duration": 0.006304,
     "end_time": "2024-11-04T20:09:35.221851",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.215547",
     "status": "completed"
    },
    "tags": [
     "parameters"
    ]
   },
   "outputs": [],
   "source": [
    "metadata_file = '/home/lauro/nupeb/rede-micro/redemicro-miliane-nutri/data/raw/metadata/miliane-metadata-CxAC.tsv'\n",
    "base_dir = os.path.join('/', 'home', 'lauro', 'nupeb', 'rede-micro', 'redemicro-miliane-nutri')\n",
    "experiment_name = 'miliane-CxAC-trim'\n",
    "class_col = 'group-id'\n",
    "replace_files = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "300bcb8a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.226483Z",
     "iopub.status.busy": "2024-11-04T20:09:35.226360Z",
     "iopub.status.idle": "2024-11-04T20:09:35.229180Z",
     "shell.execute_reply": "2024-11-04T20:09:35.228767Z"
    },
    "papermill": {
     "duration": 0.006214,
     "end_time": "2024-11-04T20:09:35.230061",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.223847",
     "status": "completed"
    },
    "tags": [
     "injected-parameters"
    ]
   },
   "outputs": [],
   "source": [
    "# Parameters\n",
    "base_dir = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri\"\n",
    "class_col = \"group-id\"\n",
    "classifier_file = \"/mnt/nupeb/rede-micro/datasets/16S_classifiers_qiime2/silva138.1_AB_V3-V4_classifier.qza\"\n",
    "experiment_name = \"ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR\"\n",
    "manifest_file = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/manifest_STD-NCxSTD-NRxHSD-NCxHSD-NR.csv\"\n",
    "metadata_file = \"/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/metadata_STD-NCxSTD-NRxHSD-NCxHSD-NR.tsv\"\n",
    "overlap = 12\n",
    "phred = 20\n",
    "replace_files = False\n",
    "threads = 6\n",
    "top_n = 20\n",
    "trim = {\"forward_primer\": \"CCTACGGGRSGCAGCAG\", \"overlap\": 8, \"reverse_primer\": \"GGACTACHVGGGTWTCTAAT\"}\n",
    "trunc_f = 0\n",
    "trunc_r = 0\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f38617d9",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.515942Z",
     "start_time": "2022-11-09T19:20:47.498345Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.235003Z",
     "iopub.status.busy": "2024-11-04T20:09:35.234638Z",
     "iopub.status.idle": "2024-11-04T20:09:35.237063Z",
     "shell.execute_reply": "2024-11-04T20:09:35.236659Z"
    },
    "papermill": {
     "duration": 0.005906,
     "end_time": "2024-11-04T20:09:35.238001",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.232095",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "experiment_folder = os.path.abspath(os.path.join(base_dir, 'experiments', experiment_name))\n",
    "img_folder = os.path.abspath(os.path.join(experiment_folder, 'imgs'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea9b90fb",
   "metadata": {
    "papermill": {
     "duration": 0.002046,
     "end_time": "2024-11-04T20:09:35.242113",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.240067",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Defining names, paths and flags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "45727eef",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.527937Z",
     "start_time": "2022-11-09T19:20:47.517696Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.246924Z",
     "iopub.status.busy": "2024-11-04T20:09:35.246737Z",
     "iopub.status.idle": "2024-11-04T20:09:35.249844Z",
     "shell.execute_reply": "2024-11-04T20:09:35.249550Z"
    },
    "papermill": {
     "duration": 0.006577,
     "end_time": "2024-11-04T20:09:35.250736",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.244159",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# QIIME2 Artifacts folder\n",
    "qiime_folder = os.path.join(experiment_folder, 'qiime-artifacts')\n",
    "\n",
    "# Input - DADA2 Artifacts\n",
    "dada2_tabs_path = os.path.join(qiime_folder, 'dada2-tabs.qza')\n",
    "dada2_reqs_path = os.path.join(qiime_folder, 'dada2-reps.qza')\n",
    "\n",
    "# PICRUST@ folder\n",
    "picrust2_folder = os.path.abspath(os.path.join(experiment_folder, 'picrust2'))\n",
    "\n",
    "# Create path if it not exist\n",
    "if not os.path.isdir(picrust2_folder):\n",
    "    os.makedirs(picrust2_folder)\n",
    "    print(f'New picrust2-artifacts folder path created: {picrust2_folder}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4903e645",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.539993Z",
     "start_time": "2022-11-09T19:20:47.529653Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.255491Z",
     "iopub.status.busy": "2024-11-04T20:09:35.255307Z",
     "iopub.status.idle": "2024-11-04T20:09:35.258768Z",
     "shell.execute_reply": "2024-11-04T20:09:35.258355Z"
    },
    "papermill": {
     "duration": 0.006893,
     "end_time": "2024-11-04T20:09:35.259734",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.252841",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Define paths for metagenome function artifacts\n",
    "ec_path = os.path.join(picrust2_folder, 'ec-pred-metagen.qza')\n",
    "ko_path = os.path.join(picrust2_folder, 'ko-pred-metagen.qza')\n",
    "pathway_path = os.path.join(picrust2_folder, 'pathway-abundance.qza')\n",
    "\n",
    "# Define paths for metagenome function visualization artifacts\n",
    "ec_viz_path = ec_path[:-1]+'v'\n",
    "ko_viz_path = ko_path[:-1]+'v'\n",
    "pathway_viz_path = pathway_path[:-1]+'v'\n",
    "\n",
    "# Define paths for biom -> tsv files (function X sample)\n",
    "pathway_fpath = os.path.join(picrust2_folder, 'pathway.tsv')\n",
    "ec_fpath = os.path.join(picrust2_folder, 'ec.tsv')\n",
    "ko_fpath = os.path.join(picrust2_folder, 'ko.tsv')\n",
    "\n",
    "# Define paths for biom -> tsv files (function X sample) - with descriptions\n",
    "pathway_desc_fpath = os.path.join(picrust2_folder, 'pathway-desc.tsv')\n",
    "ec_desc_fpath = os.path.join(picrust2_folder, 'ec-desc.tsv')\n",
    "ko_desc_fpath = os.path.join(picrust2_folder, 'ko-desc.tsv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0999a1b2",
   "metadata": {
    "papermill": {
     "duration": 0.002125,
     "end_time": "2024-11-04T20:09:35.263995",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.261870",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "## Step execution\n",
    "\n",
    "### Load input files\n",
    "\n",
    "This Step import the QIIME2 `FeatureTable[Frequency]` Artifact and the `Metadata` file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8f299d1a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.610793Z",
     "start_time": "2022-11-09T19:20:47.541540Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.268840Z",
     "iopub.status.busy": "2024-11-04T20:09:35.268715Z",
     "iopub.status.idle": "2024-11-04T20:09:35.675267Z",
     "shell.execute_reply": "2024-11-04T20:09:35.674609Z"
    },
    "papermill": {
     "duration": 0.410638,
     "end_time": "2024-11-04T20:09:35.676776",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.266138",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Load Metadata\n",
    "metadata_qa = Metadata.load(metadata_file)\n",
    "\n",
    "#Load FeatureTable[Frequency]\n",
    "tabs = Artifact.load(dada2_tabs_path)\n",
    "\n",
    "#Load FeatureTable[Sequence]\n",
    "seqs = Artifact.load(dada2_reqs_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "9b699dee",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:35.683610Z",
     "iopub.status.busy": "2024-11-04T20:09:35.683474Z",
     "iopub.status.idle": "2024-11-04T20:09:36.503690Z",
     "shell.execute_reply": "2024-11-04T20:09:36.503115Z"
    },
    "papermill": {
     "duration": 0.824191,
     "end_time": "2024-11-04T20:09:36.504798",
     "exception": false,
     "start_time": "2024-11-04T20:09:35.680607",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Filter FeatureTable[Frequency | RelativeFrequency | PresenceAbsence | Composition] based on Metadata sample ID values\n",
    "tabs = filter_samples(\n",
    "    table=tabs,\n",
    "    metadata=metadata_qa,\n",
    ").filtered_table\n",
    "# Filter SampleData[SequencesWithQuality | PairedEndSequencesWithQuality | JoinedSequencesWithQuality] based on Metadata sample ID values; returns FeatureData[Sequence | AlignedSequence]\n",
    "seqs = filter_seqs(\n",
    "    data=seqs,\n",
    "    table=tabs,\n",
    ").filtered_data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4f1b06d4",
   "metadata": {
    "papermill": {
     "duration": 0.002145,
     "end_time": "2024-11-04T20:09:36.509322",
     "exception": false,
     "start_time": "2024-11-04T20:09:36.507177",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Execute full pipelie\n",
    "\n",
    "The entire PICRUSt2 pipeline will be run using a single method, called `picrust2.methods.full_pipeline`. This method will run each of the 4 key steps: \n",
    "\n",
    "1. sequence placement\n",
    "2. hidden-state prediction of genomes\n",
    "3. metagenome prediction\n",
    "4. pathway-level predictions.\n",
    "\n",
    "More information on [Documentation](https://github.com/picrust/picrust2/wiki/Full-pipeline-script)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2c4f3874",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.642417Z",
     "start_time": "2022-11-09T19:20:47.612547Z"
    },
    "execution": {
     "iopub.execute_input": "2024-11-04T20:09:36.514421Z",
     "iopub.status.busy": "2024-11-04T20:09:36.514269Z"
    },
    "papermill": {
     "duration": 267.523724,
     "end_time": "2024-11-04T20:14:04.035169",
     "exception": false,
     "start_time": "2024-11-04T20:09:36.511445",
     "status": "completed"
    },
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1633 of 1633 sequence ids overlap between input table and FASTA.\n",
      "\n",
      "Placing sequences onto reference tree\n",
      "place_seqs.py --study_fasta /tmp/tmpe5gvh6vv/seqs.fna --ref_dir /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref --out_tree /tmp/tmpe5gvh6vv/picrust2_out/out.tre --processes 6 --intermediate /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs --min_align 0.8 --chunk_size 5000 --placement_tool sepp --verbose\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "                                              ....      ....  \n",
      "                                             '' '||.   .||'   \n",
      "                                                  ||  ||      \n",
      "                                                  '|.|'       \n",
      "     ...'   ....   ... ...  ... ...   ....        .|'|.       \n",
      "    |  ||  '' .||   ||'  ||  ||'  || '' .||      .|'  ||      \n",
      "     |''   .|' ||   ||    |  ||    | .|' ||     .|'|.  ||     \n",
      "    '....  '|..'|'. ||...'   ||...'  '|..'|.    '||'    ||:.  \n",
      "    '....'          ||       ||                               \n",
      "                   ''''     ''''   v0.8.5 (c) 2017-2024\n",
      "                                   by Lucas Czech and Pierre Barbera\n",
      "\n",
      "Invocation:                        gappa examine graft --jplace-path /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out/output_placement.json --fully-resolve --out-dir /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out\n",
      "Command:                           gappa examine graft\n",
      "\n",
      "Input:\n",
      "  --jplace-path                    /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out/output_placement.json\n",
      "\n",
      "Settings:\n",
      "  --fully-resolve                  true\n",
      "  --name-prefix\n",
      "\n",
      "Output:\n",
      "  --out-dir                        /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out\n",
      "  --file-prefix                    \n",
      "  --file-suffix\n",
      "\n",
      "Newick Tree Output:\n",
      "  --newick-tree-quote-invalid-chars\n",
      "                                   false\n",
      "\n",
      "Global Options:\n",
      "  --allow-file-overwriting         false\n",
      "  --verbose                        false\n",
      "  --threads                        10\n",
      "  --log-file\n",
      "\n",
      "Run the following command to get the references that need to be cited:\n",
      "`gappa tools citation Czech2020-genesis-and-gappa`\n",
      "\n",
      "Started 2024-11-04 20:20:32\n",
      "\n",
      "Found 1 jplace file\n",
      "\n",
      "Finished 2024-11-04 20:20:32\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "hmmalign --trim --dna --mapali /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna --informat FASTA -o /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/query_align.stockholm /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.hmm /tmp/tmpe5gvh6vv/seqs.fna\n",
      "\n",
      "Warning - 1 input sequences aligned poorly to reference sequences (--min_align option specified a minimum proportion of 0.8 aligning to reference sequences). These input sequences will not be placed and will be excluded from downstream steps.\n",
      "\n",
      "This is the set of poorly aligned input sequences to be excluded: 6fed430d2c42021ed5fa531745f38fba\n",
      "\n",
      "Raw input sequences ranged in length from 253 to 430\n",
      "\n",
      "run_sepp.py --tree /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.tre --raxml /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info --cpu 6 --molecule dna --outdir /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out -seed 297834 --alignment /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna --fragment /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/study_seqs_filtered.fasta\n",
      "[20:18:04] config.py (line 370):     INFO: Seed number: 297834\n",
      "[20:18:04] algorithm.py (line 267):     INFO: Reading input alignment: <_io.TextIOWrapper name='/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.fna' mode='r' encoding='UTF-8'>\n",
      "[20:18:05] algorithm.py (line 274):     INFO: Reading input tree: <_io.TextIOWrapper name='/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.tre' mode='r' encoding='UTF-8'>\n",
      "[20:18:05] algorithm.py (line 247):     INFO: Decomposition Sizes are set to alignment: 2000 placement: 2000\n",
      "[20:18:07] exhaustive.py (line 351):     INFO: Breaking into 16 placement subsets.\n",
      "[20:18:08] exhaustive.py (line 398):     INFO: Breaking into 16 alignment subsets.\n",
      "[20:18:08] filemgr.py (line 129):     INFO: Root temp directory built: /tmp/sepp/output.unoxcepj\n",
      "[20:18:08] exhaustive.py (line 413):     INFO: Breaking each alignment subset into 3 fragment chunks.\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_4/A_4_0/hmmbuild.input.rrim2_ei.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_0/A_0_0/hmmbuild.input.e0ijrgyx.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_3/A_3_0/hmmbuild.input.aynye2ve.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_2/A_2_0/hmmbuild.input.3tfy57tc.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta\n",
      "[20:18:10] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta\n",
      "[20:18:11] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta\n",
      "[20:18:11] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta\n",
      "[20:18:11] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta\n",
      "[20:18:11] jobs.py (line 131):     INFO: Finished hmmbuild Job with input: /tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.input.pgo2wwvi.fasta\n",
      "[20:18:19] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_4/A_4_0/hmmbuild.model.e9enuf0y, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_4/A_4_0/FC_4_0_0/hmmsearch.results.x9py8b31\n",
      "[20:18:19] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_4/A_4_0/hmmbuild.model.e9enuf0y, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_4/A_4_0/FC_4_0_1/hmmsearch.results.e9fa9uee\n",
      "[20:18:20] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_4/A_4_0/hmmbuild.model.e9enuf0y, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_4/A_4_0/FC_4_0_2/hmmsearch.results.czp9ulws\n",
      "[20:18:20] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_0/A_0_0/hmmbuild.model.pxk1laip, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_0/A_0_0/FC_0_0_0/hmmsearch.results.99whqavb\n",
      "[20:18:20] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_0/A_0_0/hmmbuild.model.pxk1laip, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_0/A_0_0/FC_0_0_2/hmmsearch.results.0ux8adhh\n",
      "[20:18:20] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_0/A_0_0/hmmbuild.model.pxk1laip, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_0/A_0_0/FC_0_0_1/hmmsearch.results.xfqbc__d\n",
      "[20:18:28] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.model.ziwu34pe, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_0/hmmsearch.results.pazm4udk\n",
      "[20:18:28] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.model.ziwu34pe, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_1/hmmsearch.results._op7m6id\n",
      "[20:18:28] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.model.ziwu34pe, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_2/hmmsearch.results.ribencnt\n",
      "[20:18:28] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.model.t8a0qbz1, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_2/hmmsearch.results.founlnw4\n",
      "[20:18:28] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.model.t8a0qbz1, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_0/hmmsearch.results.5aohxtuk\n",
      "[20:18:28] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.model.t8a0qbz1, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_1/hmmsearch.results.ll5n_g3z\n",
      "[20:18:37] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_3/A_3_0/hmmbuild.model._4a00c9p, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_3/A_3_0/FC_3_0_0/hmmsearch.results.p53zkrz5\n",
      "[20:18:37] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_3/A_3_0/hmmbuild.model._4a00c9p, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_3/A_3_0/FC_3_0_1/hmmsearch.results.o_nqw467\n",
      "[20:18:37] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_3/A_3_0/hmmbuild.model._4a00c9p, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_3/A_3_0/FC_3_0_2/hmmsearch.results.a0i7ch5n\n",
      "[20:18:37] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_2/A_2_0/hmmbuild.model.x23rqfx5, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_2/A_2_0/FC_2_0_2/hmmsearch.results.4rknqqlg\n",
      "[20:18:37] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_2/A_2_0/hmmbuild.model.x23rqfx5, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_2/A_2_0/FC_2_0_1/hmmsearch.results.4x34sh_4\n",
      "[20:18:37] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_2/A_2_0/hmmbuild.model.x23rqfx5, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_2/A_2_0/FC_2_0_0/hmmsearch.results.wqtakth2\n",
      "[20:18:46] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.model.cfl9k0vq, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_0/hmmsearch.results.urqivblj\n",
      "[20:18:46] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.model.cfl9k0vq, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_1/hmmsearch.results.bopluw6b\n",
      "[20:18:46] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.model.cfl9k0vq, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_2/hmmsearch.results.96q1h4a1\n",
      "[20:18:46] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.model.5pgdhzqk, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_1/hmmsearch.results.qysf5qrr\n",
      "[20:18:46] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.model.5pgdhzqk, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_2/hmmsearch.results.ccx770bm\n",
      "[20:18:46] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.model.5pgdhzqk, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_0/hmmsearch.results.74iz9txm\n",
      "[20:18:55] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.model.9mo8it80, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_0/hmmsearch.results.yoocc1sz\n",
      "[20:18:55] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.model.9mo8it80, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_1/hmmsearch.results.ll4aw9r7\n",
      "[20:18:55] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.model.9mo8it80, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_2/hmmsearch.results._cr5cx7i\n",
      "[20:18:55] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.model.p6w1r_ry, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_1/hmmsearch.results.aydb8fro\n",
      "[20:18:55] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.model.p6w1r_ry, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_0/hmmsearch.results.pchct7_3\n",
      "[20:18:55] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.model.p6w1r_ry, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_2/hmmsearch.results.1l7u_fg7\n",
      "[20:19:04] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.model.f95r218x, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_0/hmmsearch.results.xzp9xfru\n",
      "[20:19:04] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.model.f95r218x, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_1/hmmsearch.results.ijg7nudd\n",
      "[20:19:04] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.model.f95r218x, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_2/hmmsearch.results.20iwn00p\n",
      "[20:19:04] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.model.o8znmhdw, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_2/hmmsearch.results.cdyhkgji\n",
      "[20:19:04] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.model.o8znmhdw, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_0/hmmsearch.results.emsgct7n\n",
      "[20:19:04] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.model.o8znmhdw, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_1/hmmsearch.results.e7jytzs0\n",
      "[20:19:13] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.model.j623_ijw, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_0/hmmsearch.results.qsdxskdr\n",
      "[20:19:13] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.model.j623_ijw, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_1/hmmsearch.results.sl45oi3o\n",
      "[20:19:13] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.model.j623_ijw, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_2/hmmsearch.results.ewca6q9l\n",
      "[20:19:13] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.model.246xdwbt, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_0/hmmsearch.results._38ui29n\n",
      "[20:19:13] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.model.246xdwbt, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_2/hmmsearch.results.gi9uwssn\n",
      "[20:19:13] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.model.246xdwbt, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_1/hmmsearch.results.ldo3xq3l\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.model.mbe0fa9n, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_0/hmmsearch.results.1eyvq342\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.model.mbe0fa9n, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_1/hmmsearch.results.al_0wfum\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.model.mbe0fa9n, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_2/hmmsearch.results.j4i1dm6f\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.model.vrm0snbg, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_2ij97ccgf.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/FC_15_0_2/hmmsearch.results.bn_l52v7\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.model.vrm0snbg, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_0e7_13lk9.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/FC_15_0_0/hmmsearch.results.z3wufu8q\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmsearch Job with input: model:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.model.vrm0snbg, fragments:/tmp/sepp/output.unoxcepj/fragment_chunks/fragment_chunk_17ik023mr.fasta, elim:99999999, filter:False, output:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/FC_15_0_1/hmmsearch.results.hr92j3tb\n",
      "[20:19:22] exhaustive.py (line 81):  WARNING: Fragments [] are not scored against any subset\n",
      "[20:19:22] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_0.\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.model.ziwu34pe, fragments:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_0/hmmalign.frag.e2uje8h4.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.model.ziwu34pe, fragments:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_2/hmmalign.frag.qjm55kw_.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.model.t8a0qbz1, fragments:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_2/hmmalign.frag.fh8ppf6f.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.model.ziwu34pe, fragments:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_1/hmmalign.frag.f4zvcpc8.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta\n",
      "[20:19:22] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_2.\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.model.t8a0qbz1, fragments:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_1/hmmalign.frag.m8j4xbhc.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.model.t8a0qbz1, fragments:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_0/hmmalign.frag.8uoe15vd.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta\n",
      "[20:19:22] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_3.\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.model.5pgdhzqk, fragments:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_2/hmmalign.frag.27pfp0vx.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.model.9mo8it80, fragments:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_0/hmmalign.frag.yaf32c6r.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.model.5pgdhzqk, fragments:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_0/hmmalign.frag.fmo7m78k.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.model.9mo8it80, fragments:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_2/hmmalign.frag.zxd0k0xx.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.model.5pgdhzqk, fragments:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_1/hmmalign.frag.qjptfqfx.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.model.9mo8it80, fragments:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_1/hmmalign.frag.4wrboz76.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta\n",
      "[20:19:22] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_4.\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.model.cfl9k0vq, fragments:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_0/hmmalign.frag.qx_9vg0h.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta\n",
      "[20:19:22] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_1.\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.model.cfl9k0vq, fragments:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_1/hmmalign.frag.twyc6yz5.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta\n",
      "[20:19:22] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.model.cfl9k0vq, fragments:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_2/hmmalign.frag.k3c3dyo9.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta\n",
      "[20:19:22] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_0/hmmalign.results.9_rtyrpu) into base alignment (/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta).\n",
      "[20:19:22] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_1/hmmalign.results.e208wgu0) into base alignment (/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/FC_1_0_2/hmmalign.results.9qmsfktq) into base alignment (/tmp/sepp/output.unoxcepj/root/P_1/A_1_0/hmmbuild.input.nk9at1kw.fasta).\n",
      "[20:19:23] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_5.\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_0/hmmalign.results.qyrvw_4f) into base alignment (/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_1/hmmalign.results.r93bvfvq) into base alignment (/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/FC_5_0_2/hmmalign.results.zz4y8g83) into base alignment (/tmp/sepp/output.unoxcepj/root/P_5/A_5_0/hmmbuild.input.xg2yqbyj.fasta).\n",
      "[20:19:23] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.model.f95r218x, fragments:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_1/hmmalign.frag._j1v7wce.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta\n",
      "[20:19:23] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.model.f95r218x, fragments:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_2/hmmalign.frag.ghn0mliz.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta\n",
      "[20:19:23] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_6.\n",
      "[20:19:23] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.model.f95r218x, fragments:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_0/hmmalign.frag.o72vrzik.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_0/hmmalign.results.f7e20v0h) into base alignment (/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_1/hmmalign.results.yoafip4g) into base alignment (/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/FC_6_0_2/hmmalign.results.y4voiz_6) into base alignment (/tmp/sepp/output.unoxcepj/root/P_6/A_6_0/hmmbuild.input.3isyvnrt.fasta).\n",
      "[20:19:23] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.model.o8znmhdw, fragments:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_1/hmmalign.frag.9ugkg2ow.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta\n",
      "[20:19:23] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.model.o8znmhdw, fragments:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_2/hmmalign.frag.yy5y_3k5.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta\n",
      "[20:19:23] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.model.o8znmhdw, fragments:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_0/hmmalign.frag.5cgehd49.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta\n",
      "[20:19:23] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_7.\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_0/hmmalign.results.7jxx1xin) into base alignment (/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_1/hmmalign.results.sac12xlc) into base alignment (/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/FC_7_0_2/hmmalign.results.cs2r3zgz) into base alignment (/tmp/sepp/output.unoxcepj/root/P_7/A_7_0/hmmbuild.input.jrvltzm6.fasta).\n",
      "[20:19:23] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_8.\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_0/hmmalign.results.mjgvqff1) into base alignment (/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_1/hmmalign.results.1gn3tql7) into base alignment (/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/FC_8_0_2/hmmalign.results.wgyhm1vi) into base alignment (/tmp/sepp/output.unoxcepj/root/P_8/A_8_0/hmmbuild.input.d10frjj6.fasta).\n",
      "[20:19:23] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_10.\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_0/hmmalign.results.d407dcr4) into base alignment (/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_1/hmmalign.results.mh7yw13m) into base alignment (/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/FC_10_0_2/hmmalign.results.q12kpj0t) into base alignment (/tmp/sepp/output.unoxcepj/root/P_10/A_10_0/hmmbuild.input.ewpndmx6.fasta).\n",
      "[20:19:23] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_9.\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_0/hmmalign.results.dq9lkz17) into base alignment (/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_1/hmmalign.results.rg8agduz) into base alignment (/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta).\n",
      "[20:19:23] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/FC_9_0_2/hmmalign.results.mwak1fvi) into base alignment (/tmp/sepp/output.unoxcepj/root/P_9/A_9_0/hmmbuild.input.8211xiit.fasta).\n",
      "[20:19:25] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.model.j623_ijw, fragments:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_0/hmmalign.frag.j4fshh_p.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta\n",
      "[20:19:25] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.model.j623_ijw, fragments:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_1/hmmalign.frag.fyzeikpm.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta\n",
      "[20:19:25] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.model.j623_ijw, fragments:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_2/hmmalign.frag.vzs8jrzl.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta\n",
      "[20:19:25] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_12.\n",
      "[20:19:25] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_0/hmmalign.results.1lozqvyy) into base alignment (/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta).\n",
      "[20:19:25] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_1/hmmalign.results.8nijgt_8) into base alignment (/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta).\n",
      "[20:19:25] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/FC_12_0_2/hmmalign.results.q_6sfxvt) into base alignment (/tmp/sepp/output.unoxcepj/root/P_12/A_12_0/hmmbuild.input.4_l0lo9s.fasta).\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.model.246xdwbt, fragments:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_0/hmmalign.frag.rw0i2cq0.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.model.246xdwbt, fragments:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_2/hmmalign.frag.z276g8ra.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.model.246xdwbt, fragments:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_1/hmmalign.frag.vfu_c_gr.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta\n",
      "[20:19:26] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_13.\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_0/hmmalign.results.9fj44ns6) into base alignment (/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta).\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_1/hmmalign.results.ay94fpoc) into base alignment (/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta).\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/FC_13_0_2/hmmalign.results.eo1s5usl) into base alignment (/tmp/sepp/output.unoxcepj/root/P_13/A_13_0/hmmbuild.input.wgpn6rbo.fasta).\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.model.mbe0fa9n, fragments:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_2/hmmalign.frag.avn37qc1.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.model.mbe0fa9n, fragments:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_1/hmmalign.frag.ml3obmbp.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.model.mbe0fa9n, fragments:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_0/hmmalign.frag.0wjtmsm7.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta\n",
      "[20:19:26] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_14.\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_0/hmmalign.results.mc4qup1t) into base alignment (/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta).\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_1/hmmalign.results.6je8gztv) into base alignment (/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta).\n",
      "[20:19:26] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.model.vrm0snbg, fragments:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/FC_15_0_0/hmmalign.frag.nlncjw3s.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.input.pgo2wwvi.fasta\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/FC_14_0_2/hmmalign.results.axxnt6hy) into base alignment (/tmp/sepp/output.unoxcepj/root/P_14/A_14_0/hmmbuild.input.h53_5qjx.fasta).\n",
      "[20:19:26] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_15.\n",
      "[20:19:26] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/FC_15_0_0/hmmalign.results.q5zwxu54) into base alignment (/tmp/sepp/output.unoxcepj/root/P_15/A_15_0/hmmbuild.input.pgo2wwvi.fasta).\n",
      "[20:19:28] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.backbone.15qse5mt.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.tree.ha53amb4.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.extended.0.dnlpedgb.fasta, output:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.extended.0.dnlpedgb.jplace\n",
      "[20:19:28] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.backbone.owhcag25.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.tree.fcbitc0h.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.extended.2.a2ewj_8y.fasta, output:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.extended.2.a2ewj_8y.jplace\n",
      "[20:19:28] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.backbone.1grdfs0x.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.tree.yyhlvtsq.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.extended.1.hkf5x3uv.fasta, output:/tmp/sepp/output.unoxcepj/root/P_1/pplacer.extended.1.hkf5x3uv.jplace\n",
      "[20:19:31] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.backbone.b07yzi7h.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.tree.gkqktmtq.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.extended.2.r6dr0_qh.fasta, output:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.extended.2.r6dr0_qh.jplace\n",
      "[20:19:31] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.backbone.lenfurr7.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.tree.nztrniru.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.extended.0.j31lxhsj.fasta, output:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.extended.0.j31lxhsj.jplace\n",
      "[20:19:31] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.backbone.1oz4wh4y.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.tree.0h52fetc.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.extended.1.a2p7kb12.fasta, output:/tmp/sepp/output.unoxcepj/root/P_5/pplacer.extended.1.a2p7kb12.jplace\n",
      "[20:19:34] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.backbone.7_tu0xpd.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.tree.f9s7r169.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.extended.2.tq1dn43l.fasta, output:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.extended.2.tq1dn43l.jplace\n",
      "[20:19:34] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.model.p6w1r_ry, fragments:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_0/hmmalign.frag.79kqk_lu.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta\n",
      "[20:19:34] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.model.p6w1r_ry, fragments:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_1/hmmalign.frag.7ptvkxo5.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta\n",
      "[20:19:34] jobs.py (line 131):     INFO: Finished hmmalign Job with input: model:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.model.p6w1r_ry, fragments:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_2/hmmalign.frag.b2hgrbnh.fasta, trim:False, base_alignment:/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta\n",
      "[20:19:34] exhaustive.py (line 152):     INFO: Merging sub-alignments for placement problem : P_11.\n",
      "[20:19:34] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_0/hmmalign.results.4a0isqan) into base alignment (/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta).\n",
      "[20:19:34] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_1/hmmalign.results.6ip1w695) into base alignment (/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta).\n",
      "[20:19:34] alignment.py (line 566):     INFO: Merging extension sto file (/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/FC_11_0_2/hmmalign.results._yem8gfz) into base alignment (/tmp/sepp/output.unoxcepj/root/P_11/A_11_0/hmmbuild.input.x316k6j4.fasta).\n",
      "[20:19:36] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.backbone.9eqzcrnd.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.tree.4a82vkl4.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.extended.0.adp6gykl.fasta, output:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.extended.0.adp6gykl.jplace\n",
      "[20:19:37] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.backbone.f6xpvcsw.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.tree.msfjkpn2.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.extended.1.4ra30uor.fasta, output:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.extended.1.4ra30uor.jplace\n",
      "[20:19:37] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.backbone.5w9k60ig.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.tree.tb5k521z.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.extended.2.gnq1d0cz.fasta, output:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.extended.2.gnq1d0cz.jplace\n",
      "[20:19:37] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.backbone.umm7h_i5.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.tree.okfua2fe.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.extended.1.khx2qsid.fasta, output:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.extended.1.khx2qsid.jplace\n",
      "[20:19:37] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.backbone.6fwal24q.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.tree.qkbhdk4v.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.extended.0.3cootbd1.fasta, output:/tmp/sepp/output.unoxcepj/root/P_7/pplacer.extended.0.3cootbd1.jplace\n",
      "[20:19:39] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.backbone.b5hzwt6z.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.tree.fbb8gyyj.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.extended.1.fg6ilu3e.fasta, output:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.extended.1.fg6ilu3e.jplace\n",
      "[20:19:39] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.backbone.z0n0zfy7.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.tree.pibo8tsu.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.extended.2.bmc6jyv9.fasta, output:/tmp/sepp/output.unoxcepj/root/P_8/pplacer.extended.2.bmc6jyv9.jplace\n",
      "[20:19:39] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.backbone.unr1y_yz.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.tree.jmmfehws.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.extended.0.n1j68gsw.fasta, output:/tmp/sepp/output.unoxcepj/root/P_6/pplacer.extended.0.n1j68gsw.jplace\n",
      "[20:19:39] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.backbone.xin4rpjc.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.tree.oyhlk6ac.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.extended.2.5x3x27lb.fasta, output:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.extended.2.5x3x27lb.jplace\n",
      "[20:19:41] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.backbone.kpbw4kla.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.tree.6bhe2d1r.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.extended.1.4evn4lj0.fasta, output:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.extended.1.4evn4lj0.jplace\n",
      "[20:19:41] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.backbone.wo4vlpew.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.tree.wtp_sbvo.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.extended.0.fq12d7kp.fasta, output:/tmp/sepp/output.unoxcepj/root/P_10/pplacer.extended.0.fq12d7kp.jplace\n",
      "[20:19:43] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.backbone.46ls8dyl.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.tree.9doyp4li.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.extended.0.ldlni9pt.fasta, output:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.extended.0.ldlni9pt.jplace\n",
      "[20:19:43] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.backbone.04mjd1t_.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.tree.060nn3f0.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.extended.2.i6od6etk.fasta, output:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.extended.2.i6od6etk.jplace\n",
      "[20:19:45] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.backbone.544_cb72.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.tree.k05dnddj.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.extended.1.gwpvv8fx.fasta, output:/tmp/sepp/output.unoxcepj/root/P_9/pplacer.extended.1.gwpvv8fx.jplace\n",
      "[20:19:46] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.backbone._l5bnrqb.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.tree.xqk7e93_.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.extended.1.rd348_52.fasta, output:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.extended.1.rd348_52.jplace\n",
      "[20:19:47] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.backbone.3sw68ixq.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.tree.nbtsrx2v.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.extended.0.r8njql9j.fasta, output:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.extended.0.r8njql9j.jplace\n",
      "[20:19:48] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.backbone.f4ukf6z7.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.tree.oxwxdc4h.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.extended.2.3rql7b16.fasta, output:/tmp/sepp/output.unoxcepj/root/P_13/pplacer.extended.2.3rql7b16.jplace\n",
      "[20:19:49] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.backbone.3_3899kn.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.tree.j9bto2p9.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.extended.0.0ycgc22x.fasta, output:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.extended.0.0ycgc22x.jplace\n",
      "[20:19:50] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.backbone.oiro80ch.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.tree.3y2m0mkb.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.extended.2.7c11vowx.fasta, output:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.extended.2.7c11vowx.jplace\n",
      "[20:19:50] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.backbone.5rnb00md.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.tree.uwsq7w6w.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.extended.1.832lri51.fasta, output:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.extended.1.832lri51.jplace\n",
      "[20:19:50] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.backbone.mcd9ashg.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.tree.zkwjkdqd.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.extended.0.a6bps4y9.fasta, output:/tmp/sepp/output.unoxcepj/root/P_12/pplacer.extended.0.a6bps4y9.jplace\n",
      "[20:19:50] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.backbone.yq8ytlzv.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.tree.pcqenrx0.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.extended.2.hsl_e895.fasta, output:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.extended.2.hsl_e895.jplace\n",
      "[20:19:51] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_15/pplacer.backbone.hj99s1ec.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_15/pplacer.tree.djr6l2w5.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_15/pplacer.extended.0.som0_put.fasta, output:/tmp/sepp/output.unoxcepj/root/P_15/pplacer.extended.0.som0_put.jplace\n",
      "[20:19:52] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.backbone.bw3w5jle.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.tree.zyzyng7f.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.extended.1.355yuf_c.fasta, output:/tmp/sepp/output.unoxcepj/root/P_14/pplacer.extended.1.355yuf_c.jplace\n",
      "[20:20:25] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.backbone.rz7ij3zb.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.tree.5cs6tncm.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.extended.0.8w5vpo5d.fasta, output:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.extended.0.8w5vpo5d.jplace\n",
      "[20:20:26] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.backbone.dixxjbpl.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.tree.0c3r29pv.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.extended.1.7d6vukil.fasta, output:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.extended.1.7d6vukil.jplace\n",
      "[20:20:26] jobs.py (line 131):     INFO: Finished pplacer Job with input: backbone_alignment_file:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.backbone.ni8fkbzf.fasta, tree_file:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.tree.8i1vp9w5.tre, info_file:/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/pro_ref/pro_ref.raxml_info, extended alignment:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.extended.2.kzy5f5vv.fasta, output:/tmp/sepp/output.unoxcepj/root/P_11/pplacer.extended.2.kzy5f5vv.jplace\n",
      "[20:20:31] jobs.py (line 131):     INFO: Finished jsonmerger Job with input: input:pipe output:/tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out/output_placement.json\n",
      "[20:20:31] algorithm.py (line 196):     INFO: Current execution Finished in 147 seconds\n",
      "[20:20:31] algorithm.py (line 198):     INFO: All checkpointed executions Finished in 147 cumulative time\n",
      "[20:20:31] algorithm.py (line 207):     INFO: Temp files are left behind at /tmp/sepp/output.unoxcepj\n",
      "\n",
      "gappa examine graft --jplace-path /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out/output_placement.json --fully-resolve --out-dir /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out\n",
      "\n",
      "mv /tmp/tmpe5gvh6vv/picrust2_out/intermediate/place_seqs/sepp_out/output_placement.newick /tmp/tmpe5gvh6vv/picrust2_out/out.tre\n",
      "\n",
      "Finished placing sequences on output tree: /tmp/tmpe5gvh6vv/picrust2_out/out.tre\n",
      "hsp.py --tree /tmp/tmpe5gvh6vv/picrust2_out/out.tre --output /tmp/tmpe5gvh6vv/picrust2_out/marker_predicted_and_nsti.tsv.gz --observed_trait_table /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/16S.txt.gz --hsp_method pic --edge_exponent 0.5 --seed 100 --calculate_NSTI --processes 1 --verbose\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_nsti.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmph8g9cb9l/known_tips.txt /tmp/tmph8g9cb9l/nsti_out.txt\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp45ubzuhl/subset_tab_0 pic 0.5 FALSE FALSE /tmp/tmpkf_ix375/predicted_counts.txt /tmp/tmpkf_ix375/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "hsp.py --tree /tmp/tmpe5gvh6vv/picrust2_out/out.tre --output /tmp/tmpe5gvh6vv/picrust2_out/EC_predicted.tsv.gz --observed_trait_table /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/ec.txt.gz --hsp_method pic --edge_exponent 0.5 --seed 100 --processes 6 --verbose\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp74sfvv0w/subset_tab_1 pic 0.5 FALSE FALSE /tmp/tmp36way2dh/predicted_counts.txt /tmp/tmp36way2dh/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp74sfvv0w/subset_tab_4 pic 0.5 FALSE FALSE /tmp/tmp6ar7oay1/predicted_counts.txt /tmp/tmp6ar7oay1/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp74sfvv0w/subset_tab_2 pic 0.5 FALSE FALSE /tmp/tmpvmboot94/predicted_counts.txt /tmp/tmpvmboot94/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp74sfvv0w/subset_tab_5 pic 0.5 FALSE FALSE /tmp/tmpyf6syczy/predicted_counts.txt /tmp/tmpyf6syczy/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp74sfvv0w/subset_tab_0 pic 0.5 FALSE FALSE /tmp/tmp3nbyuq1m/predicted_counts.txt /tmp/tmp3nbyuq1m/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmp74sfvv0w/subset_tab_3 pic 0.5 FALSE FALSE /tmp/tmpuvq293zr/predicted_counts.txt /tmp/tmpuvq293zr/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "hsp.py --tree /tmp/tmpe5gvh6vv/picrust2_out/out.tre --output /tmp/tmpe5gvh6vv/picrust2_out/KO_predicted.tsv.gz --observed_trait_table /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/prokaryotic/ko.txt.gz --hsp_method pic --edge_exponent 0.5 --seed 100 --processes 6 --verbose\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_0 pic 0.5 FALSE FALSE /tmp/tmpii0p5ghp/predicted_counts.txt /tmp/tmpii0p5ghp/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_1 pic 0.5 FALSE FALSE /tmp/tmp3_swm_k7/predicted_counts.txt /tmp/tmp3_swm_k7/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_2 pic 0.5 FALSE FALSE /tmp/tmp03uw8l80/predicted_counts.txt /tmp/tmp03uw8l80/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_3 pic 0.5 FALSE FALSE /tmp/tmppzs97_z3/predicted_counts.txt /tmp/tmppzs97_z3/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_4 pic 0.5 FALSE FALSE /tmp/tmpyaf2bhmo/predicted_counts.txt /tmp/tmpyaf2bhmo/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_5 pic 0.5 FALSE FALSE /tmp/tmpco83o1zo/predicted_counts.txt /tmp/tmpco83o1zo/predicted_ci.txt 100\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_6 pic 0.5 FALSE FALSE /tmp/tmpfz_8b837/predicted_counts.txt /tmp/tmpfz_8b837/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_7 pic 0.5 FALSE FALSE /tmp/tmpopxcoc49/predicted_counts.txt /tmp/tmpopxcoc49/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_8 pic 0.5 FALSE FALSE /tmp/tmpdwe_ubuj/predicted_counts.txt /tmp/tmpdwe_ubuj/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_9 pic 0.5 FALSE FALSE /tmp/tmprqdg_6s9/predicted_counts.txt /tmp/tmprqdg_6s9/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_10 pic 0.5 FALSE FALSE /tmp/tmpcur7079n/predicted_counts.txt /tmp/tmpcur7079n/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_11 pic 0.5 FALSE FALSE /tmp/tmpnxhcggdw/predicted_counts.txt /tmp/tmpnxhcggdw/predicted_ci.txt 100\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_12 pic 0.5 FALSE FALSE /tmp/tmpkjidnbz8/predicted_counts.txt /tmp/tmpkjidnbz8/predicted_ci.txt 100\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_13 pic 0.5 FALSE FALSE /tmp/tmphl7nya69/predicted_counts.txt /tmp/tmphl7nya69/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_14 pic 0.5 FALSE FALSE /tmp/tmpo5hyhwx0/predicted_counts.txt /tmp/tmpo5hyhwx0/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_15 pic 0.5 FALSE FALSE /tmp/tmpz1iyuf_b/predicted_counts.txt /tmp/tmpz1iyuf_b/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_16 pic 0.5 FALSE FALSE /tmp/tmpf3a0hgas/predicted_counts.txt /tmp/tmpf3a0hgas/predicted_ci.txt 100\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_17 pic 0.5 FALSE FALSE /tmp/tmpqm_8vuuv/predicted_counts.txt /tmp/tmpqm_8vuuv/predicted_ci.txt 100\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_18 pic 0.5 FALSE FALSE /tmp/tmp0mvlaqp8/predicted_counts.txt /tmp/tmp0mvlaqp8/predicted_ci.txt 100\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_19 pic 0.5 FALSE FALSE /tmp/tmp1jja129k/predicted_counts.txt /tmp/tmp1jja129k/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_20 pic 0.5 FALSE FALSE /tmp/tmpwwdgkbqn/predicted_counts.txt /tmp/tmpwwdgkbqn/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "Rscript /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/Rscripts/castor_hsp.R /tmp/tmpe5gvh6vv/picrust2_out/out.tre /tmp/tmpy0n2ntfn/subset_tab_21 pic 0.5 FALSE FALSE /tmp/tmp61l0spwt/predicted_counts.txt /tmp/tmp61l0spwt/predicted_ci.txt 100\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Running metagenome pipeline for EC\n",
      "metagenome_pipeline.py --input /tmp/tmpe5gvh6vv/intable.biom --function /tmp/tmpe5gvh6vv/picrust2_out/EC_predicted.tsv.gz --min_reads 1 --min_samples 1 --out_dir /tmp/tmpe5gvh6vv/picrust2_out/EC_metagenome_out --max_nsti 2 --marker /tmp/tmpe5gvh6vv/picrust2_out/marker_predicted_and_nsti.tsv.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1 of 1632 ASVs were above the max NSTI cut-off of 2.0 and were removed from the downstream analyses.\n",
      "\n",
      "Running metagenome pipeline for KO\n",
      "metagenome_pipeline.py --input /tmp/tmpe5gvh6vv/intable.biom --function /tmp/tmpe5gvh6vv/picrust2_out/KO_predicted.tsv.gz --min_reads 1 --min_samples 1 --out_dir /tmp/tmpe5gvh6vv/picrust2_out/KO_metagenome_out --max_nsti 2 --marker /tmp/tmpe5gvh6vv/picrust2_out/marker_predicted_and_nsti.tsv.gz\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Inferring pathways from predicted EC\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1 of 1632 ASVs were above the max NSTI cut-off of 2.0 and were removed from the downstream analyses.\n",
      "\n",
      "pathway_pipeline.py --input /tmp/tmpe5gvh6vv/picrust2_out/EC_metagenome_out/pred_metagenome_unstrat.tsv.gz --out_dir /tmp/tmpe5gvh6vv/picrust2_out/pathways_out --map /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/pathway_mapfiles/metacyc_path2rxn_struc_filt_pro.txt --intermediate /tmp/tmpe5gvh6vv/picrust2_out/intermediate/pathways --proc 6 --regroup_map /opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/picrust2/default_files/pathway_mapfiles/ec_level4_to_metacyc_rxn.tsv --verbose\n",
      "\n",
      "Wrote predicted pathway abundances and coverages to /tmp/tmpe5gvh6vv/picrust2_out/pathways_out\n"
     ]
    }
   ],
   "source": [
    "need_pipeline = replace_files\n",
    "need_pipeline |= not (os.path.isfile(ec_path) and os.path.isfile(ko_path) and os.path.isfile(pathway_path))\n",
    "if need_pipeline:\n",
    "    results = picrust2.methods.full_pipeline(\n",
    "        table=tabs, \n",
    "        seq=seqs, \n",
    "        threads=6, \n",
    "        placement_tool='sepp',\n",
    "        hsp_method='pic', \n",
    "        max_nsti=2,\n",
    "        highly_verbose=True\n",
    "    )\n",
    "    ec_metagenome = results.ec_metagenome\n",
    "    ko_metagenome = results.ko_metagenome\n",
    "    pathway_abundance = results.pathway_abundance\n",
    "else:\n",
    "    ec_metagenome = Artifact.load(ec_path)\n",
    "    ko_metagenome = Artifact.load(ko_path)\n",
    "    pathway_abundance = Artifact.load(pathway_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "771308c4",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Persist created artifacts\n",
    "\n",
    "We will define file paths and persist all artifacts. We start with `.qza` files. We will save the visualization files in sequence as `qzv` files. Finally, we save a `biom`-like file as `tsv` with brief descriptions of all functions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c680ab46",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.646444Z",
     "start_time": "2022-11-09T19:20:47.643806Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "if need_pipeline:\n",
    "    # Export artifact folder\n",
    "    ec_metagenome.export_data(output_dir=ec_path.split('.')[0])\n",
    "    ko_metagenome.export_data(output_dir=ko_path.split('.')[0])\n",
    "    pathway_abundance.export_data(output_dir=pathway_path.split('.')[0])\n",
    "    \n",
    "    # Save artifacts as .qza files\n",
    "    ec_metagenome.save(ec_path)\n",
    "    ko_metagenome.save(ko_path)\n",
    "    pathway_abundance.save(pathway_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d61ebec3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.687022Z",
     "start_time": "2022-11-09T19:20:47.647637Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/seaborn/_oldcore.py:1119: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n",
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/seaborn/_oldcore.py:1119: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n",
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/seaborn/_oldcore.py:1119: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n",
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/seaborn/_oldcore.py:1119: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n",
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/seaborn/_oldcore.py:1119: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n",
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/seaborn/_oldcore.py:1119: FutureWarning: use_inf_as_na option is deprecated and will be removed in a future version. Convert inf values to NaN before operating instead.\n",
      "  with pd.option_context('mode.use_inf_as_na', True):\n"
     ]
    }
   ],
   "source": [
    "need_viz = replace_files\n",
    "need_viz |= not (os.path.isfile(ec_viz_path) and os.path.isfile(ko_viz_path) and os.path.isfile(pathway_viz_path))\n",
    "if need_viz:\n",
    "    # Create visualization artifacts\n",
    "    ec_viz = summarize(table=ec_metagenome, sample_metadata=metadata_qa).visualization\n",
    "    ko_viz = summarize(table=ko_metagenome, sample_metadata=metadata_qa).visualization\n",
    "    path_viz = summarize(table=pathway_abundance, sample_metadata=metadata_qa).visualization\n",
    "\n",
    "    # Save visualization artifacts as .qzv files\n",
    "    ec_viz.save(ec_viz_path)\n",
    "    ko_viz.save(ko_viz_path)\n",
    "    path_viz.save(pathway_viz_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fb471714",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing ec\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/skbio/util/_warning.py:60: UserWarning: observed_otus is deprecated as of 0.6.0.\n",
      "  warn(f\"{func.__name__} is deprecated as of {ver}.\")\n",
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/q2_diversity_lib/skbio/_methods.py:71: RuntimeWarning: invalid value encountered in scalar divide\n",
      "  return _shannon(counts, base=np.e) / np.log(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing ko\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/q2_diversity_lib/skbio/_methods.py:71: RuntimeWarning: invalid value encountered in scalar divide\n",
      "  return _shannon(counts, base=np.e) / np.log(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing path\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/envs/qiime2-amplicon-2024.5/lib/python3.9/site-packages/q2_diversity_lib/skbio/_methods.py:71: RuntimeWarning: invalid value encountered in scalar divide\n",
      "  return _shannon(counts, base=np.e) / np.log(\n"
     ]
    }
   ],
   "source": [
    "tables = (ec_metagenome, ko_metagenome, pathway_abundance)\n",
    "table_types = ('ec', 'ko', 'path')\n",
    "\n",
    "for i, t in enumerate(tables):\n",
    "    print(f'Processing {table_types[i]}')\n",
    "    results = core_metrics(\n",
    "        table = t,\n",
    "        sampling_depth = 1,\n",
    "        metadata = metadata_qa,\n",
    "        n_jobs = 6,\n",
    "    )\n",
    "    emperor_out = os.path.join(picrust2_folder, f'{table_types[i]}-bray_curtis_emperor.qzv')\n",
    "    results.bray_curtis_emperor.save(emperor_out)\n",
    "    emperor_out = os.path.join(picrust2_folder, f'{table_types[i]}-jaccard_emperor.qzv')\n",
    "    results.jaccard_emperor.save(emperor_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "57f1f4d8",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-11-09T19:20:47.704885Z",
     "start_time": "2022-11-09T19:20:47.689312Z"
    },
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "need_biom = replace_files\n",
    "need_biom |= not (os.path.isfile(ec_fpath) and os.path.isfile(ko_fpath) and os.path.isfile(pathway_fpath))\n",
    "if need_biom:\n",
    "    # Convert artifact to dataframe - transposed\n",
    "    df_pathway = pathway_abundance.view(pd.DataFrame).T\n",
    "    df_ec = ec_metagenome.view(pd.DataFrame).T\n",
    "    df_ko = ko_metagenome.view(pd.DataFrame).T\n",
    "\n",
    "    # Write dataframe to a tsv file\n",
    "    df_pathway.to_csv(pathway_fpath, sep='\\t', index=True)\n",
    "    df_ec.to_csv(ec_fpath, sep='\\t', index=True)\n",
    "    df_ko.to_csv(ko_fpath, sep='\\t', index=True)\n",
    "    \n",
    "    # Put a new column with descriptions of each function's ID\n",
    "    !add_descriptions.py -i {ec_fpath} -m EC -o {ec_desc_fpath}\n",
    "    !add_descriptions.py -i {ko_fpath} -m KO -o {ko_desc_fpath}\n",
    "    !add_descriptions.py -i {pathway_fpath} -m METACYC -o {pathway_desc_fpath}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9b05c8b",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "### Aggregate by sum groups counts\n",
    "\n",
    "We create new columns with the aggregation of the counts for all groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d6384378",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Create a new DataFrame with Metadata info\n",
    "metadata_df = metadata_qa.to_dataframe()\n",
    "# Select all groups\n",
    "group_ids = metadata_df[class_col].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "da1b5dbd",
   "metadata": {
    "papermill": {
     "duration": null,
     "end_time": null,
     "exception": null,
     "start_time": null,
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Iterate over all PICRUST2 files\n",
    "for f in (ec_desc_fpath, ko_desc_fpath, pathway_desc_fpath):\n",
    "    # Create new DataFrame to store new data\n",
    "    df = pd.read_csv(f, sep='\\t')\n",
    "    new_df = pd.DataFrame()\n",
    "    # Iterate over all groups\n",
    "    for g in group_ids:\n",
    "        # Select all IDs from current group\n",
    "        idx = metadata_df[metadata_df[class_col]==g].index\n",
    "        # Sum all values from samples of the current group\n",
    "        tmp = df.loc[:, idx].sum(axis=1)\n",
    "        # Create a new column with the sum\n",
    "        new_df[g] = tmp\n",
    "    # Sum the values of all groups\n",
    "    new_df['Total'] = new_df.sum(axis=1)\n",
    "    # Join new columns into the end of the original DataFrame\n",
    "    new_df = pd.concat([df, new_df])\n",
    "    # Write joined DataFrame to file - overwrite\n",
    "    new_df.to_csv(f, index=False, sep='\\t')"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 275.106769,
   "end_time": "2024-11-04T20:14:04.465282",
   "environment_variables": {},
   "exception": null,
   "input_path": "nb-templates/step-picrust2-analysis.ipynb",
   "output_path": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/experiments/ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR/nb-executed-steps/step-picrust2-analysis-ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR.ipynb",
   "parameters": {
    "base_dir": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri",
    "class_col": "group-id",
    "classifier_file": "/mnt/nupeb/rede-micro/datasets/16S_classifiers_qiime2/silva138.1_AB_V3-V4_classifier.qza",
    "experiment_name": "ana-flavia-STD-NCxSTD-NRxHSD-NCxHSD-NR",
    "manifest_file": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/manifest_STD-NCxSTD-NRxHSD-NCxHSD-NR.csv",
    "metadata_file": "/mnt/nupeb/rede-micro/redemicro-ana-flavia-nutri/data/raw/selected_groups/metadata_STD-NCxSTD-NRxHSD-NCxHSD-NR.tsv",
    "overlap": 12,
    "phred": 20,
    "replace_files": false,
    "threads": 6,
    "top_n": 20,
    "trim": {
     "forward_primer": "CCTACGGGRSGCAGCAG",
     "overlap": 8,
     "reverse_primer": "GGACTACHVGGGTWTCTAAT"
    },
    "trunc_f": 0,
    "trunc_r": 0
   },
   "start_time": "2024-11-04T20:09:29.358513",
   "version": "2.3.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
